diff --git a/distribution/src/bin/opensearch-env b/distribution/src/bin/opensearch-env
index cd7a0b2b452..a09db73e9f4 100644
--- a/distribution/src/bin/opensearch-env
+++ b/distribution/src/bin/opensearch-env
@@ -46,6 +46,10 @@ fi
 # now set the classpath
 OPENSEARCH_CLASSPATH="$OPENSEARCH_HOME/lib/*"
 
+if [[ -n "${JAVA_FIPS_OPTIONS}" ]]; then
+    OPENSEARCH_CLASSPATH="${JAVA_FIPS_OPTIONS}:${OPENSEARCH_CLASSPATH}"
+fi
+
 # now set the path to java: OPENSEARCH_JAVA_HOME -> JAVA_HOME -> bundled JRE -> bundled JDK
 if [ ! -z "$OPENSEARCH_JAVA_HOME" ]; then
   JAVA="$OPENSEARCH_JAVA_HOME/bin/java"
