diff --git a/plugins/repository-azure/build.gradle b/plugins/repository-azure/build.gradle
index c9297bd5..9e22a821 100644
--- a/plugins/repository-azure/build.gradle
+++ b/plugins/repository-azure/build.gradle
@@ -68,7 +68,7 @@ dependencies {
   api 'com.nimbusds:lang-tag:1.7'
   // Both msal4j:1.14.3 and oauth2-oidc-sdk:11.9.1 has compile dependency on different versions of json-smart,
   // selected the higher version which is 2.5.0
-  api 'net.minidev:json-smart:2.5.1'
+  api 'net.minidev:json-smart:2.5.2'
   api 'net.minidev:accessors-smart:2.5.1'
   api "org.ow2.asm:asm:${versions.asm}"
   // End of transitive dependencies for azure-identity
diff --git a/plugins/repository-hdfs/build.gradle b/plugins/repository-hdfs/build.gradle
index c2685a52..e2a4d3eb 100644
--- a/plugins/repository-hdfs/build.gradle
+++ b/plugins/repository-hdfs/build.gradle
@@ -81,7 +81,7 @@ dependencies {
   api 'javax.servlet:servlet-api:2.5'
   api "org.slf4j:slf4j-api:${versions.slf4j}"
   api "org.apache.logging.log4j:log4j-slf4j-impl:${versions.log4j}"
-  api 'net.minidev:json-smart:2.5.1'
+  api 'net.minidev:json-smart:2.5.2'
   api "io.netty:netty-all:${versions.netty}"
   implementation "com.fasterxml.woodstox:woodstox-core:${versions.woodstox}"
   implementation 'org.codehaus.woodstox:stax2-api:4.2.2'
