From 23b00ef9f82f17b05a54e49089da7844017e3de4 Mon Sep 17 00:00:00 2001
From: Dentrax <furkan.turkal@chainguard.dev>
Date: Sun, 23 Feb 2025 19:24:52 +0300
Subject: [PATCH] GHSA-4g8c-wm8x-jfhw

Signed-off-by: Dentrax <furkan.turkal@chainguard.dev>
---
 build.gradle          | 6 ++++++
 buildSrc/build.gradle | 1 +
 2 files changed, 7 insertions(+)

diff --git a/build.gradle b/build.gradle
index a0a253fc..b9d4d163 100644
--- a/build.gradle
+++ b/build.gradle
@@ -328,6 +328,12 @@ allprojects {
     javadoc.options.addStringOption("-release", java.targetCompatibility.majorVersion)
   }
 
+  configurations.all {
+    resolutionStrategy {
+      force 'io.netty:netty-handler:4.1.118.Final'
+    }
+  }
+
   // support for reproducible builds
   tasks.withType(AbstractArchiveTask).configureEach { task ->
     // ignore file timestamps
diff --git a/buildSrc/build.gradle b/buildSrc/build.gradle
index 44bbd95e..4be3d872 100644
--- a/buildSrc/build.gradle
+++ b/buildSrc/build.gradle
@@ -137,6 +137,7 @@ dependencies {
 configurations.all {
   resolutionStrategy {
     force "com.google.guava:guava:${props.getProperty('guava')}"
+    force 'io.netty:netty-handler:4.1.118.Final'
   }
 }
 
-- 
2.39.5 (Apple Git-154)

