commit af0118d13e52f5a08841464a768e01a0bf3e3075
Author: Aidyn-A <31858918+Aidyn-A@users.noreply.github.com>
Date:   Sun Aug 3 11:43:07 2025 +0400

    use cudaDeviceGetAttribute

diff --git a/tensorpipe/channel/cuda_ipc/context_impl.cc b/tensorpipe/channel/cuda_ipc/context_impl.cc
index 69db487..b4c1fda 100644
--- a/tensorpipe/channel/cuda_ipc/context_impl.cc
+++ b/tensorpipe/channel/cuda_ipc/context_impl.cc
@@ -233,7 +233,9 @@ std::shared_ptr<ContextImpl> ContextImpl::create() {
 
     // The other two compute modes are "exclusive" and "prohibited", both of
     // which prevent access from an other process.
-    if (props.computeMode != cudaComputeModeDefault) {
+    int computeMode = -1;
+    TP_CUDA_CHECK(cudaDeviceGetAttribute(&computeMode, cudaDevAttrComputeMode, device.index));
+    if (computeMode != cudaComputeModeDefault) {
       TP_VLOG(4) << "CUDA IPC channel is not viable because CUDA device "
                  << device.index << " is not in default compute mode";
       return nullptr;
