From 0dcc6f58209f5b4f59d693cbb792743e2ac88e4e Mon Sep 17 00:00:00 2001
From: Philippe Deslauriers <philde@chainguard.dev>
Date: Sat, 11 Nov 2023 14:54:59 -0800
Subject: [PATCH] Patch to enable bumping grpc

Signed-off-by: Philippe Deslauriers <philde@chainguard.dev>
---
 pkg/server/gateway_builder.go | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/pkg/server/gateway_builder.go b/pkg/server/gateway_builder.go
index 9182f8e..cd69ac2 100644
--- a/pkg/server/gateway_builder.go
+++ b/pkg/server/gateway_builder.go
@@ -19,17 +19,17 @@ import (
 	"crypto/tls"
 	"crypto/x509"
 	"fmt"
+	"net"
+	"time"
+
 	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
 	retry "github.com/grpc-ecosystem/go-grpc-middleware/retry"
 	grpc_prometheus "github.com/grpc-ecosystem/go-grpc-prometheus"
 	pb "github.com/uswitch/kiam/proto"
 	"google.golang.org/grpc"
-	"google.golang.org/grpc/balancer/roundrobin"
 	"google.golang.org/grpc/credentials"
 	"google.golang.org/grpc/keepalive"
 	"google.golang.org/grpc/security/advancedtls"
-	"net"
-	"time"
 )
 
 const (
@@ -121,7 +121,7 @@ func (b *KiamGatewayBuilder) Build(ctx context.Context) (*KiamGateway, error) {
 				retry.WithBackoff(retry.BackoffLinear(b.retryInterval)),
 			),
 		)),
-		grpc.WithBalancerName(roundrobin.Name),
+		grpc.WithDefaultServiceConfig(`{"loadBalancingPolicy":"round_robin"}`),
 		grpc.WithDisableServiceConfig(),
 		grpc.WithBlock(),
 		grpc.WithStreamInterceptor(grpc_prometheus.StreamClientInterceptor),
-- 
2.42.0

