diff --git a/azurelinuxagent/ga/cgroupapi.py b/azurelinuxagent/ga/cgroupapi.py
index b030633f..f4ff31f2 100644
--- a/azurelinuxagent/ga/cgroupapi.py
+++ b/azurelinuxagent/ga/cgroupapi.py
@@ -67,7 +67,8 @@ class CGroupUtil(object):
         except ValueError:
             return False
         return (distro_name.lower() == 'ubuntu' and distro_version.major >= 16) or \
-               (distro_name.lower() in ('centos', 'redhat') and 8 <= distro_version.major < 9)
+               (distro_name.lower() in ('centos', 'redhat') and 8 <= distro_version.major < 9) or \
+               distro_name.lower() == 'chainguard'
 
     @staticmethod
     def get_extension_slice_name(extension_name, old_slice=False):
