diff --git a/common/CudaWorker/DcgmDgemm.cpp b/common/CudaWorker/DcgmDgemm.cpp
index 8d33a32..09d6779 100644
--- a/common/CudaWorker/DcgmDgemm.cpp
+++ b/common/CudaWorker/DcgmDgemm.cpp
@@ -15,6 +15,7 @@
  */
 #include "DcgmDgemm.hpp"
 
+#include <cstdint>
 #include <exception>
 #include <stdexcept>
 
@@ -122,4 +123,4 @@ cublasStatus_t DcgmDgemm(cublasLtHandle_t ltHandle,
     return CUBLAS_STATUS_SUCCESS;
 }
 
-} // namespace DcgmNs
\ No newline at end of file
+} // namespace DcgmNs
diff --git a/common/DcgmError.h b/common/DcgmError.h
index 8638cdc..4f78870 100644
--- a/common/DcgmError.h
+++ b/common/DcgmError.h
@@ -15,6 +15,7 @@
  */
 #pragma once
 
+#include <cstdint>
 #include <sstream>
 #include <string>
 
diff --git a/common/DcgmStringHelpers.cpp b/common/DcgmStringHelpers.cpp
index b41917e..dc6aa9b 100644
--- a/common/DcgmStringHelpers.cpp
+++ b/common/DcgmStringHelpers.cpp
@@ -15,6 +15,7 @@
  */
 #include "DcgmStringHelpers.h"
 
+#include <algorithm>
 #include <cstring>
 #include <string>
 
diff --git a/dcgmi/CommandOutputController.cpp b/dcgmi/CommandOutputController.cpp
index 5057205..443008e 100644
--- a/dcgmi/CommandOutputController.cpp
+++ b/dcgmi/CommandOutputController.cpp
@@ -25,6 +25,7 @@
 #include <DcgmStringHelpers.h>
 #include <algorithm>
 #include <cstdarg>
+#include <functional>
 #include <iostream>
 #include <string>
 
diff --git a/dcgmi/Diag.h b/dcgmi/Diag.h
index e0aa259..62cfa1d 100644
--- a/dcgmi/Diag.h
+++ b/dcgmi/Diag.h
@@ -24,6 +24,7 @@
 #define DIAG_H_
 
 #include <optional>
+#include <functional>
 
 #include "Command.h"
 #include "CommandOutputController.h"
diff --git a/dcgmlib/src/DcgmGpuInstance.cpp b/dcgmlib/src/DcgmGpuInstance.cpp
index 8696249..108c11c 100755
--- a/dcgmlib/src/DcgmGpuInstance.cpp
+++ b/dcgmlib/src/DcgmGpuInstance.cpp
@@ -13,6 +13,8 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#include <algorithm>
+
 #include "DcgmGpuInstance.h"
 
 #include <DcgmLogging.h>
diff --git a/hostengine/src/HostEngineOutput.cpp b/hostengine/src/HostEngineOutput.cpp
index 23c6ca9..798b83b 100644
--- a/hostengine/src/HostEngineOutput.cpp
+++ b/hostengine/src/HostEngineOutput.cpp
@@ -20,6 +20,7 @@
 #include <iostream>
 #include <limits>
 #include <string_view>
+#include <unordered_map>
 
 namespace
 {
@@ -365,4 +366,4 @@ void HostEngineOutput::PrintLongUsage(TCLAP::CmdLineInterface &cmdLine, std::ost
         }
     }
     os << std::endl;
-}
\ No newline at end of file
+}
diff --git a/nvvs/src/NvvsCommon.cpp b/nvvs/src/NvvsCommon.cpp
index 8f78886..54dcc3e 100644
--- a/nvvs/src/NvvsCommon.cpp
+++ b/nvvs/src/NvvsCommon.cpp
@@ -13,6 +13,8 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+#include <algorithm>
 #include <sstream>
 #include <stdexcept>
 #include <sys/stat.h>
diff --git a/nvvs/src/TestFramework.cpp b/nvvs/src/TestFramework.cpp
index 241277d..183e065 100644
--- a/nvvs/src/TestFramework.cpp
+++ b/nvvs/src/TestFramework.cpp
@@ -25,6 +25,7 @@
 #include <TestFramework.h>
 
 #include <atomic>
+#include <algorithm>
 #include <cerrno>
 #include <cstdio>
 #include <cstdlib>
diff --git a/sdk/nvidia/nvml/nvml_loader/nvml_loader.cpp b/sdk/nvidia/nvml/nvml_loader/nvml_loader.cpp
index 9eebeaf..ab06439 100644
--- a/sdk/nvidia/nvml/nvml_loader/nvml_loader.cpp
+++ b/sdk/nvidia/nvml/nvml_loader/nvml_loader.cpp
@@ -19,6 +19,7 @@
 #include <dlfcn.h>
 
 #include <atomic>
+#include <cstdlib>
 #include <mutex>
 
 static void *g_nvmlLib                                     = 0;
