--- a/libstdc++-v3/src/nonshared17/cow-fs_path.cc
+++ b/libstdc++-v3/src/nonshared17/cow-fs_path.cc
@@ -91,7 +91,7 @@
 asm (".hidden _ZNSs9_M_mutateEmmm");
 asm (".hidden _ZNSs6insertEmPKcm");
 asm (".hidden _ZNSbIwSt11char_traitsIwESaIwEE9_M_mutateEmmm");
-asm (".hidden _ZSt16__do_str_codecvtISswSt7codecvtIwc11__mbstate_tES1_MS2_KFNSt12codecvt_base6resultERS1_PKwS7_RS7_PcS9_RS9_EEbPKT0_SF_RT_RKT1_RT2_RmT3_");
+//asm (".hidden _ZSt16__do_str_codecvtISswSt7codecvtIwc11__mbstate_tES1_MS2_KFNSt12codecvt_base6resultERS1_PKwS7_RS7_PcS9_RS9_EEbPKT0_SF_RT_RKT1_RT2_RmT3_");
 asm (".hidden _ZNSsC2ERKSsmm");
 asm (".hidden _ZNSsC1ERKSsmm");
 asm (".hidden _ZNSbIwSt11char_traitsIwESaIwEE7reserveEm");
