diff --git a/melange-out/keycloak-bitnami-fips/opt/bitnami/scripts/keycloak/entrypoint.sh b/melange-out/keycloak-bitnami-fips/opt/bitnami/scripts/keycloak/entrypoint.sh
index 188dab62ae4..2c1e1b89051 100755
--- a/melange-out/keycloak-bitnami-fips/opt/bitnami/scripts/keycloak/entrypoint.sh
+++ b/melange-out/keycloak-bitnami-fips/opt/bitnami/scripts/keycloak/entrypoint.sh
@@ -9,6 +9,10 @@ set -o nounset
 set -o pipefail
 #set -o xtrace # Uncomment this line for debugging purposes
 
+# Copy BCFIPS to providers
+mkdir -p /opt/bitnami/keycloak/providers
+cp -r /usr/share/java/bouncycastle-fips/* /opt/bitnami/keycloak/providers/
+
 # Load libraries
 . /opt/bitnami/scripts/libbitnami.sh
 . /opt/bitnami/scripts/liblog.sh
