diff --git a/integration/client-cli/admin-cli/src/main/bin/kcadm.sh b/integration/client-cli/admin-cli/src/main/bin/kcadm.sh
index fa0e7f86a5..0de3076fce 100755
--- a/integration/client-cli/admin-cli/src/main/bin/kcadm.sh
+++ b/integration/client-cli/admin-cli/src/main/bin/kcadm.sh
@@ -28,4 +28,9 @@ if [ -z "$JAVA" ]; then
     fi
 fi
 
-exec "$JAVA" $KC_OPTS -cp $DIRNAME/client/keycloak-admin-cli-${project.version}.jar --add-opens=java.base/java.security=ALL-UNNAMED -Dkc.lib.dir=$DIRNAME/client/lib org.keycloak.client.admin.cli.KcAdmMain "$@"
+# If CLASSPATH is not set, fall back to default JAR path
+if [ -z "$CLASSPATH" ]; then
+    CLASSPATH="$DIRNAME/client/keycloak-admin-cli-${project.version}.jar"
+fi
+
+exec "$JAVA" $KC_OPTS -cp "$CLASSPATH" --add-opens=java.base/java.security=ALL-UNNAMED -Dkc.lib.dir=$DIRNAME/client/lib org.keycloak.client.registration.cli.KcRegMain "$@"
diff --git a/integration/client-cli/admin-cli/src/main/bin/kcreg.sh b/integration/client-cli/admin-cli/src/main/bin/kcreg.sh
index be65bcadb1..bf46cad9ae 100755
--- a/integration/client-cli/admin-cli/src/main/bin/kcreg.sh
+++ b/integration/client-cli/admin-cli/src/main/bin/kcreg.sh
@@ -27,4 +27,10 @@ if [ -z "$JAVA" ]; then
 fi
 
 DIRNAME="$(dirname "$RESOLVED_NAME")"
-exec "$JAVA" $KC_OPTS -cp $DIRNAME/client/keycloak-admin-cli-${project.version}.jar --add-opens=java.base/java.security=ALL-UNNAMED -Dkc.lib.dir=$DIRNAME/client/lib org.keycloak.client.registration.cli.KcRegMain "$@"
+
+# If CLASSPATH is not set, fall back to default JAR path
+if [ -z "$CLASSPATH" ]; then
+    CLASSPATH="$DIRNAME/client/keycloak-admin-cli-${project.version}.jar"
+fi
+
+exec "$JAVA" $KC_OPTS -cp "$CLASSPATH" --add-opens=java.base/java.security=ALL-UNNAMED -Dkc.lib.dir=$DIRNAME/client/lib org.keycloak.client.registration.cli.KcRegMain "$@"
