diff --git a/pkg/reconciliation/reconcile_datacenter.go b/pkg/reconciliation/reconcile_datacenter.go
index d9ff927..de5a435 100644
--- a/pkg/reconciliation/reconcile_datacenter.go
+++ b/pkg/reconciliation/reconcile_datacenter.go
@@ -88,11 +88,6 @@ func (rc *ReconciliationContext) ProcessDeletion() result.ReconcileResult {
 		rc.ReqLogger.Error(err, "Failed to remove dynamic secret watches for CassandraDatacenter")
 	}
 
-	if err := rc.deletePVCs(); err != nil {
-		rc.ReqLogger.Error(err, "Failed to delete PVCs for CassandraDatacenter")
-		return result.Error(err)
-	}
-
 	// Update finalizer to allow delete of CassandraDatacenter
 	rc.Datacenter.SetFinalizers(nil)
 	rc.Datacenter.Spec.Size = origSize // Has to be set to original size, since 0 isn't allowed for the Update to succeed
