diff --git a/pkg/provider/docker/docker.go b/pkg/provider/docker/docker.go
index 0516620ce..1463590cc 100644
--- a/pkg/provider/docker/docker.go
+++ b/pkg/provider/docker/docker.go
@@ -309,7 +309,7 @@ func (p *Provider) Provide(configurationChan chan<- dynamic.Message, pool *safe.
 						case event := <-eventsc:
 							if event.Action == "start" ||
 								event.Action == "die" ||
-								strings.HasPrefix(event.Action, "health_status") {
+								strings.HasPrefix(string(event.Action), "health_status") {
 								startStopHandle(event)
 							}
 						case err := <-errc:
