diff --git a/pyproject.toml b/pyproject.toml
index 9468224..9a45857 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -55,24 +55,12 @@ dev = [
 packages = [
         "pulp",
         "pulp.solverdir",
-        "pulp.solverdir.cbc.linux.i32",
-        "pulp.solverdir.cbc.linux.i64",
-        "pulp.solverdir.cbc.linux.arm64",
-        "pulp.solverdir.cbc.win.i32",
-        "pulp.solverdir.cbc.win.i64",
-        "pulp.solverdir.cbc.osx.i64",
         "pulp.apis",
         "pulp.tests"]
 # This is a workaround for https://github.com/astral-sh/uv/issues/9513
 license-files = []
 
 [tool.setuptools.package-data]
-"pulp.solverdir.cbc.linux.i32"= ["*", "*.*"]
-"pulp.solverdir.cbc.linux.i64"= ["*", "*.*"]
-"pulp.solverdir.cbc.linux.arm64"= ["*", "*.*"]
-"pulp.solverdir.cbc.win.i32"= ["*", "*.*"]
-"pulp.solverdir.cbc.win.i64"= ["*", "*.*"]
-"pulp.solverdir.cbc.osx.i64"= ["*", "*.*"]
 
 [project.scripts]
 pulptest = "pulp.tests.run_tests:pulpTestAll"
