diff --git a/open-vm-tools/lib/include/guest_os.h b/open-vm-tools/lib/include/guest_os.h
index 6af88c45d..1beead8ee 100644
--- a/open-vm-tools/lib/include/guest_os.h
+++ b/open-vm-tools/lib/include/guest_os.h
@@ -324,6 +324,7 @@ Bool Gos_InSetArray(uint32 gos, const uint32 *set);
 #define STR_OS_AUROX               "Aurox"
 #define STR_OS_BLACKCAT            "BlackCat"
 #define STR_OS_CENTOS              "centos"
+#define STR_OS_CHAINGUARD          "Chainguard"
 #define STR_OS_CRXPOD              "CRXPod"
 #define STR_OS_CRXSYS              "CRXSys"
 #define STR_OS_COBALT              "Cobalt"
@@ -393,6 +394,7 @@ Bool Gos_InSetArray(uint32 gos, const uint32 *set);
 #define STR_OS_ULTRAPENGUIN       "UltraPenguin"
 #define STR_OS_UNITEDLINUX        "UnitedLinux"
 #define STR_OS_VALINUX            "VALinux"
+#define STR_OS_WOLFI              "Wolfi"
 #define STR_OS_YELLOW_DOG         "Yellow Dog"
 #define STR_OS_ECOMSTATION        "eComStation"
 
diff --git a/open-vm-tools/lib/misc/hostinfoPosix.c b/open-vm-tools/lib/misc/hostinfoPosix.c
index 542195776..548dc9876 100644
--- a/open-vm-tools/lib/misc/hostinfoPosix.c
+++ b/open-vm-tools/lib/misc/hostinfoPosix.c
@@ -1408,6 +1408,7 @@ static const ShortNameSet shortNameArray[] = {
 /* Big Cloud must come before Red Hat Entry */
 { "black cat",           STR_OS_BLACKCAT,           HostinfoGenericSetShortName },
 { "centos",              NULL,                      HostinfoSetCentosShortName  },
+{ "chainguard",          STR_OS_CHAINGUARD,         HostinfoGenericSetShortName },
 { "cobalt",              STR_OS_COBALT,             HostinfoGenericSetShortName },
 { "conectiva",           STR_OS_CONECTIVA,          HostinfoGenericSetShortName },
 { "debian",              NULL,                      HostinfoSetDebianShortName  },
@@ -1442,6 +1443,7 @@ static const ShortNameSet shortNameArray[] = {
 { "united linux",        STR_OS_UNITEDLINUX,        HostinfoGenericSetShortName },
 { "va linux",            STR_OS_VALINUX,            HostinfoGenericSetShortName },
 { "vmware photon",       STR_OS_PHOTON,             HostinfoGenericSetShortName },
+{ "wolfi",               STR_OS_WOLFI,              HostinfoGenericSetShortName },
 { "yellow dog",          STR_OS_YELLOW_DOG,         HostinfoGenericSetShortName },
 { NULL,                  NULL,                      NULL                        } // MUST BE LAST
 };
-- 
2.48.1

