From afa0fe45033483979bf453c8fe32b0c302e0ab88 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Thu, 21 Aug 2025 18:19:05 +0100
Subject: [PATCH] Add chainguard patches

---
 WORKSPACE | 1 +
 1 file changed, 1 insertion(+)

diff --git a/WORKSPACE b/WORKSPACE
index 861ea3a4..de9abd23 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -44,6 +44,7 @@ git_repository(
         "@//patches:0005-thread_local-reset-slot-in-worker-threads-first.patch",
         "@//patches:0006-http-header-expose-attribute.patch",
         "@//patches:0007-liburing-arm-build.patch",
+        "//:0001-version-Print-FIPS-module-version.patch",
     ],
     # // clang-format off: Envoy's format check: Only repository_locations.bzl may contains URL references
     remote = "https://github.com/envoyproxy/envoy.git",
-- 
2.48.1

