--- a/deps/Makefile
+++ b/deps/Makefile
@@ -31,7 +31,6 @@ targets :=	check_openssl_version \
 			lz4 \
 			cityhash \
 			microhttpd \
-			curl \
 			ev \
 			libhttpserver \
 			libinjection \
--- a/include/makefiles_paths.mk
+++ b/include/makefiles_paths.mk
@@ -74,9 +74,9 @@ COREDUMPER_PATH := $(DEPS_PATH)/coredumper/coredumper
 COREDUMPER_IDIR := $(COREDUMPER_PATH)/include
 COREDUMPER_LDIR := $(COREDUMPER_PATH)/src

-CURL_PATH := $(DEPS_PATH)/curl/curl
-CURL_IDIR := $(CURL_PATH)/include
-CURL_LDIR := $(CURL_PATH)/lib/.libs
+# Use system curl
+CURL_IDIR := /usr/include
+CURL_LDIR := /usr/lib

 EV_PATH := $(DEPS_PATH)/libev/libev
 EV_IDIR := $(EV_PATH)
--- a/src/Makefile
+++ b/src/Makefile
@@ -91,7 +91,6 @@ STATICMYLIBS := -Wl,-Bstatic \
 				-lhttpserver \
 				-lmicrohttpd \
 				-linjection \
-				-lcurl \
 				-lev \
 				-lscram \
 				-lusual \
@@ -105,7 +104,7 @@ ifeq ($(UNAME_S),Linux)
 	STATICMYLIBS += -lcoredumper
 endif

-MYLIBS := -Wl,--export-dynamic $(STATICMYLIBS) -Wl,-Bdynamic -lpthread -lssl -lcrypto -lm -lz -lrt -lprometheus-cpp-pull -lprometheus-cpp-core -luuid $(EXTRALINK)
+MYLIBS := -Wl,--export-dynamic $(STATICMYLIBS) -Wl,-Bdynamic -lcurl -lpthread -lssl -lcrypto -lm -lz -lrt -lprometheus-cpp-pull -lprometheus-cpp-core -luuid $(EXTRALINK)
 ifeq ($(DISTRO),almalinux)
 ifeq ($(CENTOSVER),8)
 	MYLIBS := -Wl,--export-dynamic $(STATICMYLIBS) -Wl,-Bdynamic -lpthread $(LIB_SSL_PATH) $(LIB_CRYPTO_PATH) -lm -lz -lrt -lprometheus-cpp-pull -lprometheus-cpp-core -luuid $(EXTRALINK)
