--- ./backend/tensorrt_llm/scripts/build_wheel.py.orig
+++ ./backend/tensorrt_llm/scripts/build_wheel.py
@@ -85,8 +85,8 @@
     )
 
     # Ensure compatible virtualenv version is installed (>=20.29.1, <22.0)
-    print("-- Ensuring virtualenv version >=20.29.1,<22.0 is installed...")
-    build_run(f'"{sys.executable}" -m pip install "virtualenv>=20.29.1,<22.0"')
+    #print("-- Ensuring virtualenv version >=20.29.1,<22.0 is installed...")
+    #build_run(f'"{sys.executable}" -m pip install "virtualenv>=20.29.1,<22.0"')
 
     # Create venv if it doesn't exist
     if not venv_dir.exists():
@@ -105,7 +105,7 @@
     print(
         f"-- Installing requirements from {requirements_file} into {venv_dir}..."
     )
-    build_run(f'"{venv_python}" -m pip install -r "{requirements_file}"')
+    #build_run(f'"{venv_python}" -m pip install -r "{requirements_file}"')
 
     venv_conan = setup_conan(scripts_dir, venv_python)
 
@@ -113,7 +113,7 @@
 
 
 def setup_conan(scripts_dir, venv_python):
-    build_run(f'"{venv_python}" -m pip install conan==2.14.0')
+    #build_run(f'"{venv_python}" -m pip install conan==2.14.0')
     # Determine the path to the conan executable within the venv
     venv_conan = scripts_dir / "conan"
     if not venv_conan.exists():
@@ -525,8 +525,8 @@
 
         install_file(get_pybind_lib(), pkg_dir)
         if not skip_stubs:
-            with working_directory(project_dir):
-                build_run(f"\"{venv_python}\" -m pip install pybind11-stubgen")
+            #with working_directory(project_dir):
+            #    build_run(f"\"{venv_python}\" -m pip install pybind11-stubgen")
             with working_directory(pkg_dir):
                 if on_windows:
                     stubgen = "stubgen.py"
@@ -588,7 +588,7 @@
         )
 
     if install:
-        build_run(f"\"{sys.executable}\" -m pip install -e .[devel]")
+        pass #build_run(f"\"{sys.executable}\" -m pip install -e .[devel]")
 
 
 def add_arguments(parser: ArgumentParser):
