From cd190e1bf02cb8812809b4cb558d46b3fb51e162 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Wed, 20 Aug 2025 16:22:01 +0100
Subject: [PATCH] Add envoy patches

Patch istio, to add envoy patches.
---

diff --git a/WORKSPACE b/WORKSPACE
index e639876..bb06a95 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -37,6 +37,8 @@ http_archive(
     name = "envoy",
     patch_args = ["-p1"],
     patches = [
+        "//:0001-version-Print-FIPS-module-version.patch",
+        "//:0001-Add-quiche.patch",
         "@io_istio_proxy//bazel:0001-tls-fix-SAN-validation-for-OTHERNAME-types-with-embe.patch",
         "@io_istio_proxy//bazel:0002-fix-jwt_auth-crash-with-two-or-more-auth-header.patch",
         "@io_istio_proxy//bazel:0003-Add-option-to-reject-early-CONNECT-data.patch",
