diff --git a/src/dev/build/tasks/build_packages_task.ts b/src/dev/build/tasks/build_packages_task.ts
index 99984d54a..08911acd1 100644
--- a/src/dev/build/tasks/build_packages_task.ts
+++ b/src/dev/build/tasks/build_packages_task.ts
@@ -318,6 +318,7 @@ export async function buildWebpackBundles({
   noCache?: boolean;
 }) {
   const options = [
+    ['--allow-root'],
     quiet ? ['--quiet'] : [],
     dist ? ['--dist'] : [],
     noCache ? ['--no-cache'] : [],
