From 3eeff5900b3e1d2fda601cf8fc237bf819abc57d Mon Sep 17 00:00:00 2001
From: Srishti Hegde <srishti.hegde@chainguard.dev>
Date: Fri, 1 Nov 2024 17:25:04 -0700
Subject: [PATCH] patches tls cipher in consul

---
 share/cluster/consul.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/share/cluster/consul.go b/share/cluster/consul.go
index a080cb5b..e1f103ce 100644
--- a/share/cluster/consul.go
+++ b/share/cluster/consul.go
@@ -172,7 +172,7 @@ func createConfigFile(cc *ClusterConfig) error {
 	sa = append(sa, fmt.Sprintf("        \"serf_lan\": %d,\n", lanPort))
 	sa = append(sa, fmt.Sprintf("        \"serf_wan\": %d\n", -1))
 	sa = append(sa, fmt.Sprintf("    },\n"))
-	sa = append(sa, fmt.Sprintf("    \"tls_cipher_suites\": \"TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384\",\n"))
+	sa = append(sa, fmt.Sprintf("    \"tls_cipher_suites\": \"TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384\",\n"))
 	sa = append(sa, fmt.Sprintf("    \"performance\": {\n"))
 	sa = append(sa, fmt.Sprintf("        \"rpc_hold_timeout\": \"%ds\"\n", 300))
 	sa = append(sa, fmt.Sprintf("    }\n"))
-- 
2.39.5 (Apple Git-154)

