diff --git a/cloudinit/config/cc_keyboard.py b/cloudinit/config/cc_keyboard.py
index 8657ab4..166a661 100644
--- a/cloudinit/config/cc_keyboard.py
+++ b/cloudinit/config/cc_keyboard.py
@@ -21,7 +21,7 @@ from cloudinit.settings import PER_INSTANCE
 DEFAULT_KEYBOARD_MODEL = "pc105"
 
 supported_distros = distros.Distro.expand_osfamily(
-    ["alpine", "arch", "debian", "redhat", "suse"]
+    ["alpine", "arch", "chainguard", "debian", "redhat", "suse"]
 )
 
 meta: MetaSchema = {
diff --git a/cloudinit/config/cc_resizefs.py b/cloudinit/config/cc_resizefs.py
index f7d74f1..2e50069 100644
--- a/cloudinit/config/cc_resizefs.py
+++ b/cloudinit/config/cc_resizefs.py
@@ -213,6 +213,11 @@ def maybe_get_writable_device_path(devpath, info):
         )
         devpath = fulldevpath
 
+    # I don't want to teach cloud-init to deal with PARTLABEL it's not meant for it anyway.
+    if "PARTLABEL" in devpath:
+        LOG.debug("Ignoring root with PARTLABEL")
+        return None
+
     try:
         statret = os.stat(devpath)
     except OSError as exc:
diff --git a/cloudinit/config/cc_ssh_import_id.py b/cloudinit/config/cc_ssh_import_id.py
index 7d98832..1243488 100644
--- a/cloudinit/config/cc_ssh_import_id.py
+++ b/cloudinit/config/cc_ssh_import_id.py
@@ -23,7 +23,7 @@ SSH_IMPORT_ID_BINARY = "ssh-import-id"
 
 meta: MetaSchema = {
     "id": "cc_ssh_import_id",
-    "distros": ["alpine", "cos", "debian", "raspberry-pi-os", "ubuntu"],
+    "distros": ["alpine", "cos", "chainguard", "debian", "raspberry-pi-os", "ubuntu"],
     "frequency": PER_INSTANCE,
     "activate_by_schema_keys": [],
 }
diff --git a/cloudinit/distros/__init__.py b/cloudinit/distros/__init__.py
index eb58095..4de910f 100644
--- a/cloudinit/distros/__init__.py
+++ b/cloudinit/distros/__init__.py
@@ -64,6 +64,7 @@ OSFAMILIES = {
     "alpine": ["alpine"],
     "aosc": ["aosc"],
     "arch": ["arch"],
+    "chainguard": ["chainguard"],
     "debian": ["debian", "ubuntu", "raspberry-pi-os"],
     "freebsd": ["freebsd", "dragonfly"],
     "gentoo": ["gentoo", "cos"],
@@ -97,6 +98,7 @@ OSFAMILIES = {
     ],
     "openeuler": ["openeuler"],
     "OpenCloudOS": ["OpenCloudOS", "TencentOS"],
+    "Chainguard": ["Chainguard"] # TODO: Maybe wolfi or chainguard wolfi?
 }
 
 LOG = logging.getLogger(__name__)
diff --git a/cloudinit/distros/chainguard.py b/cloudinit/distros/chainguard.py
new file mode 100644
index 0000000..a5ec8c9
--- /dev/null
+++ b/cloudinit/distros/chainguard.py
@@ -0,0 +1,143 @@
+# Copyright (C) 2025  Justin Vreeland
+#
+# Author: Justin Vreeland <justin.vreeland@chainguard.dev>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import logging
+import os
+import re
+import stat
+from datetime import datetime
+from typing import Any, Dict, Optional
+
+from cloudinit import distros, helpers, lifecycle, subp, util
+from cloudinit.distros.parsers.hostname import HostnameConf
+from cloudinit.settings import PER_ALWAYS, PER_INSTANCE
+
+LOG = logging.getLogger(__name__)
+
+# TODO: see if we can disable the cloud-init growparts/resizefs by default
+# it gets grumpy about out cmdline args
+#
+# "Device '/dev/PARTLABEL=Chainguard' did not exist. cannot resize: dev=/dev/root mnt_point=/ path=/",
+
+class Distro(distros.Distro):
+    pip_package_name = "py3-pip"
+    init_cmd = ["systemctl"]
+    dhclient_lease_directory = "/var/lib/dhcp"
+    dhclient_lease_file_regex = r"dhclient\.leases"
+
+    def __init__(self, name, cfg, paths):
+        distros.Distro.__init__(self, name, cfg, paths)
+        # This will be used to restrict certain
+        # calls from repeatedly happening (when they
+        # should only happen say once per instance...)
+        self._runner = helpers.Runners(paths)
+        self.default_locale = "C.UTF-8"
+        self.osfamily = "chainguard"
+        cfg["ssh_svcname"] = "sshd"
+
+    def install_packages(self, pkglist: distros.PackageList):
+        self.update_package_sources()
+        self.package_command("add", pkgs=pkglist)
+
+    def _write_hostname(self, hostname, filename):
+        conf = None
+        try:
+            # Try to update the previous one
+            # so lets see if we can read it first.
+            conf = self._read_hostname_conf(filename)
+        except IOError:
+            create_hostname_file = util.get_cfg_option_bool(
+                self._cfg, "create_hostname_file", True
+            )
+            if create_hostname_file:
+                pass
+            else:
+                LOG.info(
+                    "create_hostname_file is False; hostname file not created"
+                )
+                return
+        if not conf:
+            conf = HostnameConf("")
+        conf.set_hostname(hostname)
+        util.write_file(filename, str(conf), 0o644)
+
+    def _read_system_hostname(self):
+        sys_hostname = self._read_hostname(self.hostname_conf_fn)
+        return (self.hostname_conf_fn, sys_hostname)
+
+    def _read_hostname_conf(self, filename):
+        conf = HostnameConf(util.load_text_file(filename))
+        conf.parse()
+        return conf
+
+    def _read_hostname(self, filename, default=None):
+        hostname = None
+        try:
+            conf = self._read_hostname_conf(filename)
+            hostname = conf.hostname
+        except IOError:
+            pass
+        if not hostname:
+            return default
+        return hostname
+
+    def _get_localhost_ip(self):
+        return "127.0.1.1"
+
+    def apply_locale(self, locale, out_fn=None):
+        # Some other distros are doing some handling here
+        # for regenerating locales or printing logs or whatnot
+        # We pre-generate our locales into different packages
+        # localectl should update this file and systemd can
+        # regenrate locales if we need to
+        # Maybe we should update the exports in /etc/profile.d as well?
+        # I really think that should just be removed from the VMs though.
+        # It seems like it's there due to alpine musl/no systemd stuff
+        # and because containers lack full infrustrtucture around those things
+        # If we're using systemd the locale should just be set when we build the image
+        # instead of in that export.
+        #
+        # Some distro seem more diverse than other about this.
+        # I didn't find docs explingin how this is supposed to work
+        # we don't support other locale configuration locations and
+        # i'm not going to write random file locations. out_fn
+        subp.subp(["localectl", "set-locale", locale], capture=False)
+
+    def set_timezone(self, tz):
+        distros.set_etc_timezone(tz=tz, tz_file=self._find_tz_file(tz))
+
+    def package_command(self, command, args=None, pkgs=None):
+        if pkgs is None:
+            pkgs = []
+
+        cmd = ["apk"]
+        # Redirect output
+        cmd.append("--quiet")
+
+        if args and isinstance(args, str):
+            cmd.append(args)
+        elif args and isinstance(args, list):
+            cmd.extend(args)
+
+        if command:
+            cmd.append(command)
+
+        if command == "upgrade":
+            cmd.extend(["--update-cache", "--available"])
+
+        pkglist = util.expand_package_list("%s-%s", pkgs)
+        cmd.extend(pkglist)
+
+        # Allow the output of this to flow outwards (ie not be captured)
+        subp.subp(cmd, capture=False)
+
+    def update_package_sources(self, *, force=False):
+        self._runner.run(
+            "update-sources",
+            self.package_command,
+            ["update"],
+            freq=PER_ALWAYS if force else PER_INSTANCE,
+        )
diff --git a/cloudinit/util.py b/cloudinit/util.py
index 348cd86..87a9550 100644
--- a/cloudinit/util.py
+++ b/cloudinit/util.py
@@ -617,6 +617,7 @@ def _get_variant(info):
             "arch",
             "azurelinux",
             "centos",
+            "chainguard",
             "cloudlinux",
             "debian",
             "eurolinux",
@@ -635,6 +636,8 @@ def _get_variant(info):
             "virtuozzo",
         ):
             variant = linux_dist
+        elif linux_dist in ("wolfi",):
+            variant = "chainguard"
         elif linux_dist in ("ubuntu", "linuxmint", "mint"):
             variant = "ubuntu"
         elif linux_dist == "redhat":
diff --git a/config/cloud.cfg.tmpl b/config/cloud.cfg.tmpl
index dea7f03..f5b7c5c 100644
--- a/config/cloud.cfg.tmpl
+++ b/config/cloud.cfg.tmpl
@@ -14,6 +14,7 @@
                  "ubuntu": "Ubuntu", "unknown": "Ubuntu"}) %}
 {% set groups = ({"alpine": "adm, wheel", "aosc": "wheel", "arch": "wheel, users",
                   "azurelinux": "wheel",
+                  "chainguard": "adm,  wheel",
                   "debian": "adm, audio, cdrom, dialout, dip, floppy, netdev, plugdev, sudo, video",
                   "gentoo": "users, wheel", "mariner": "wheel",
                   "photon": "wheel",
@@ -233,7 +234,7 @@ cloud_final_modules:
 # (not accessible to handlers/transforms)
 system_info:
   # This will affect which distro class gets used
-{% if variant in ["alpine", "amazon", "aosc", "arch", "azurelinux", "debian", "fedora",
+{% if variant in ["alpine", "amazon", "aosc", "arch", "azurelinux", "chainguard", "debian", "fedora",
                   "freebsd", "gentoo", "mariner", "netbsd", "openbsd",
                   "OpenCloudOS", "openeuler", "openmandriva", "photon",
                   "raspberry-pi-os", "suse", "TencentOS", "ubuntu"] or is_rhel %}
@@ -251,7 +252,7 @@ system_info:
 {% else %}
     name: {{ variant }}
 {% endif %}
-{% if variant in ["alpine", "amazon", "aosc", "arch", "azurelinux", "debian", "fedora",
+{% if variant in ["alpine", "amazon", "aosc", "arch", "chainguard", "azurelinux", "debian", "fedora",
                   "gentoo", "mariner", "OpenCloudOS", "openeuler",
                   "openmandriva", "photon", "raspberry-pi-os",
                   "suse", "TencentOS", "ubuntu", "unknown"]
@@ -339,7 +340,7 @@ system_info:
 {% elif variant == "raspberry-pi-os" %}
   ntp_client: 'systemd-timesyncd'
 {% endif %}
-{% if variant in ["alpine", "amazon", "aosc", "arch", "azurelinux", "debian", "fedora",
+{% if variant in ["alpine", "amazon", "aosc", "arch", "chainguard", "azurelinux", "debian", "fedora",
                   "gentoo", "mariner", "OpenCloudOS", "openeuler",
                   "openmandriva", "photon", "suse", "TencentOS", "ubuntu",
                   "unknown"]
@@ -400,7 +401,7 @@ system_info:
 {% endif %}
 {% if variant in ["debian", "raspberry-pi-os", "ubuntu", "unknown"] %}
   ssh_svcname: ssh
-{% elif variant in ["alpine", "amazon", "aosc", "arch", "azurelinux", "fedora",
+{% elif variant in ["alpine", "amazon", "aosc", "arch", "azurelinux", "chainguard", "fedora",
                     "gentoo", "mariner", "OpenCloudOS", "openeuler",
                     "openmandriva", "photon", "suse", "TencentOS"]
                    or is_rhel %}
diff --git a/tools/render-template b/tools/render-template
index 4b5efcf..260b3c4 100755
--- a/tools/render-template
+++ b/tools/render-template
@@ -20,6 +20,7 @@ def main():
         "benchmark",
         "centos",
         "cloudlinux",
+        "chainguard",
         "debian",
         "dragonfly",
         "eurolinux",
