diff -u b/Gemfile b/Gemfile
--- b/Gemfile
+++ b/Gemfile
@@ -94,7 +94,7 @@
 gem 'doorkeeper', '~> 5.8', '>= 5.8.1', feature_category: :system_access
 gem 'doorkeeper-openid_connect', '~> 1.8.10', feature_category: :system_access
 gem 'doorkeeper-device_authorization_grant', '~> 1.0.0', feature_category: :system_access
-gem 'rexml', '~> 3.4.0', feature_category: :shared
+gem 'rexml', '~> 3.4.2', feature_category: :shared
 gem 'ruby-saml', '~> 1.18', feature_category: :system_access
 gem 'omniauth-saml', '~> 2.2.1', feature_category: :system_access
 gem 'omniauth', '~> 2.1.0', feature_category: :system_access
@@ -147,7 +147,7 @@
 gem 'ohai', '~> 18.1', feature_category: :product_analytics
 
 # GPG
-gem 'gpgme', '~> 2.0.24', feature_category: :source_code_management
+gem 'gpgme', '~> 2.0.25', feature_category: :source_code_management
 
 # LDAP Auth
 # GitLab fork with several improvements to original library. For full list of changes
diff -u b/Gemfile.lock b/Gemfile.lock
--- b/Gemfile.lock
+++ b/Gemfile.lock
@@ -1651,7 +1651,7 @@
     retriable (3.1.2)
     reverse_markdown (3.0.0)
       nokogiri
-    rexml (3.4.1)
+    rexml (3.4.2)
     rinku (2.0.0)
     rotp (6.3.0)
     rouge (4.6.0)
@@ -2202,7 +2202,7 @@
   google-cloud-storage (~> 1.45.0)
   google-protobuf (~> 3.25, >= 3.25.3)
   googleauth (~> 1.14)
-  gpgme (~> 2.0.24)
+  gpgme (~> 2.0.25)
   grape (~> 2.0.0)
   grape-entity (~> 1.0.1)
   grape-path-helpers (~> 2.0.1)
