From 3fa9cdb2f67c4a8da0f182893845cee05f285e28 Mon Sep 17 00:00:00 2001
From: Batuhan Apaydin <batuhan.apaydin@chainguard.dev>
Date: Mon, 6 Nov 2023 20:31:40 +0300
Subject: [PATCH] disableWerror

Signed-off-by: Batuhan Apaydin <batuhan.apaydin@chainguard.dev>
---
 SConstruct | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SConstruct b/SConstruct
index f572ceae250..5df3da5451d 100644
--- a/SConstruct
+++ b/SConstruct
@@ -1576,7 +1576,7 @@ detectSystem.Finish()
 if env.TargetOSIs('posix'):
     if env.ToolchainIs('gcc', 'clang'):
         env.Append(
-            CCFLAGS_WERROR=["-Werror"],
+            # CCFLAGS_WERROR=["-Werror"],
             CXXFLAGS_WERROR=['-Werror=unused-result'] if env.ToolchainIs('clang') else [],
             LINKFLAGS_WERROR=['-Wl,-fatal_warnings' if env.TargetOSIs('darwin') else "-Wl,--fatal-warnings"],
         )
-- 
2.39.3 (Apple Git-145)

