diff --git a/src/mongo/util/net/ssl_options_server.idl b/src/mongo/util/net/ssl_options_server.idl
index 4e95c357..0c3de510 100644
--- a/src/mongo/util/net/ssl_options_server.idl
+++ b/src/mongo/util/net/ssl_options_server.idl
@@ -154,13 +154,6 @@ configs:
     deprecated_short_name: sslAllowInvalidCertificates
     arg_vartype: Switch
     cpp_varname: sslGlobalParams.sslAllowInvalidCertificates
-  "net.tls.FIPSMode":
-    description: "Activate FIPS 140-2 mode at startup"
-    short_name: tlsFIPSMode
-    deprecated_name: "net.ssl.FIPSMode"
-    deprecated_short_name: sslFIPSMode
-    arg_vartype: Switch
-    cpp_varname: sslGlobalParams.sslFIPSMode

   # Certificate Selectors are only available on OSX/Windows with --ssl-provider=native (or auto)
   "net.tls.certificateSelector":
diff --git a/src/mongo/util/net/ssl_options_test.cpp b/src/mongo/util/net/ssl_options_test.cpp
index 76ad7f48..48b49509 100644
--- a/src/mongo/util/net/ssl_options_test.cpp
+++ b/src/mongo/util/net/ssl_options_test.cpp
@@ -221,7 +221,6 @@ TEST(SetupOptions, tlsModeRequired) {
     argv.push_back("--tlsAllowInvalidHostnames");
     argv.push_back("--tlsAllowInvalidCertificates");
     argv.push_back("--tlsWeakCertificateValidation");
-    argv.push_back("--tlsFIPSMode");
     argv.push_back("--tlsCertificateKeyFilePassword");
     argv.push_back("pw1");
     argv.push_back("--tlsClusterPassword");
@@ -251,7 +250,6 @@ TEST(SetupOptions, tlsModeRequired) {
     ASSERT_EQ(::mongo::sslGlobalParams.sslAllowInvalidHostnames, true);
     ASSERT_EQ(::mongo::sslGlobalParams.sslAllowInvalidCertificates, true);
     ASSERT_EQ(::mongo::sslGlobalParams.sslWeakCertificateValidation, true);
-    ASSERT_EQ(::mongo::sslGlobalParams.sslFIPSMode, true);
     ASSERT_EQ(::mongo::sslGlobalParams.sslPEMKeyPassword, "pw1");
     ASSERT_EQ(::mongo::sslGlobalParams.sslClusterPassword, "pw2");
     ASSERT_EQ(static_cast<int>(::mongo::sslGlobalParams.sslDisabledProtocols.back()),
@@ -287,7 +285,6 @@ TEST(SetupOptions, sslModeRequired) {
     argv.push_back("--sslAllowInvalidHostnames");
     argv.push_back("--sslAllowInvalidCertificates");
     argv.push_back("--sslWeakCertificateValidation");
-    argv.push_back("--sslFIPSMode");
     argv.push_back("--sslPEMKeyPassword");
     argv.push_back("pw1");
     argv.push_back("--sslClusterPassword");
@@ -317,7 +314,6 @@ TEST(SetupOptions, sslModeRequired) {
     ASSERT_EQ(::mongo::sslGlobalParams.sslAllowInvalidHostnames, true);
     ASSERT_EQ(::mongo::sslGlobalParams.sslAllowInvalidCertificates, true);
     ASSERT_EQ(::mongo::sslGlobalParams.sslWeakCertificateValidation, true);
-    ASSERT_EQ(::mongo::sslGlobalParams.sslFIPSMode, true);
     ASSERT_EQ(::mongo::sslGlobalParams.sslPEMKeyPassword, "pw1");
     ASSERT_EQ(::mongo::sslGlobalParams.sslClusterPassword, "pw2");
     ASSERT_EQ(static_cast<int>(::mongo::sslGlobalParams.sslDisabledProtocols.back()),
