From aef65d3bb775a3d056a041c309d2dae6fd47756a Mon Sep 17 00:00:00 2001
From: Sanjay Kumar <sanjay.kumar@chainguard.dev>
Date: Thu, 5 Feb 2026 12:17:24 +0530
Subject: [PATCH] md5 to sha256 internal worker

---
 internal/internal_worker.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/internal/internal_worker.go b/internal/internal_worker.go
index fa3f2af..dd7417d 100644
--- a/internal/internal_worker.go
+++ b/internal/internal_worker.go
@@ -25,7 +25,7 @@ package internal
 
 import (
 	"context"
-	"crypto/md5"
+	"crypto/sha256"
 	"encoding/hex"
 	"errors"
 	"fmt"
@@ -955,7 +955,7 @@ func initBinaryChecksum() (string, error) {
 		_ = f.Close() // error is unimportant as it is read-only
 	}()
 
-	h := md5.New()
+	h := sha256.New()
 	if _, err := io.Copy(h, f); err != nil {
 		return "", err
 	}
-- 
2.50.1 (Apple Git-155)

