From aa3d717653471e1037542d63c34e7cdc7de4c647 Mon Sep 17 00:00:00 2001
From: Sanjay Kumar <sanjay.kumar@chainguard.dev>
Date: Thu, 5 Feb 2026 00:46:28 +0530
Subject: [PATCH] md5 to sha256

---
 tools/common/schema/updatetask.go | 11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

diff --git a/tools/common/schema/updatetask.go b/tools/common/schema/updatetask.go
index 778a256de..2bfa522a5 100644
--- a/tools/common/schema/updatetask.go
+++ b/tools/common/schema/updatetask.go
@@ -21,10 +21,7 @@
 package schema
 
 import (
-	// In this context md5 is just used for versioning the current schema. It is a weak cryptographic primitive and
-	// should not be used for anything more important (password hashes etc.). Marking it as #nosec because of how it's
-	// being used.
-	"crypto/md5" // #nosec
+	"crypto/sha256"
 	"encoding/hex"
 	"encoding/json"
 	"fmt"
@@ -315,10 +312,8 @@ func readManifest(fileSystem fs.FS, subdir string) (*manifest, error) {
 		return nil, fmt.Errorf("manifest missing SchemaUpdateCqlFiles")
 	}
 
-	// See comment above. This is an appropriate usage of md5.
-	// #nosec
-	md5Bytes := md5.Sum(jsonBlob)
-	manifest.md5 = hex.EncodeToString(md5Bytes[:])
+	hashBytes := sha256.Sum256(jsonBlob)
+	manifest.md5 = hex.EncodeToString(hashBytes[:])
 
 	return &manifest, nil
 }
-- 
2.50.1 (Apple Git-155)

