diff --git a/dev/make-distribution.sh b/dev/make-distribution.sh
index 948ee19fbac..2a587627f09 100755
--- a/dev/make-distribution.sh
+++ b/dev/make-distribution.sh
@@ -166,7 +166,7 @@ export MAVEN_OPTS="${MAVEN_OPTS:--Xss128m -Xmx4g -XX:ReservedCodeCacheSize=128m}
 # Store the command as an array because $MVN variable might have spaces in it.
 # Normal quoting tricks don't work.
 # See: http://mywiki.wooledge.org/BashFAQ/050
-BUILD_COMMAND=("$MVN" clean package -DskipTests $@)
+BUILD_COMMAND=("$MVN" -T$(grep -c processor /proc/cpuinfo) clean package -DskipTests $@)

 # Actually build the jar
 echo -e "\nBuilding with..."
