--- a/bindings/go/CMakeLists.txt
+++ b/bindings/go/CMakeLists.txt
@@ -129,7 +129,7 @@ endfunction()
 
-build_go_package(LIBRARY NAME fdb_go PATH fdb INCLUDE_TEST)
+build_go_package(LIBRARY NAME fdb_go PATH fdb)
 add_dependencies(fdb_go fdb_c go_options_file)
 
-build_go_package(LIBRARY NAME tuple_go PATH fdb/tuple INCLUDE_TEST)
+build_go_package(LIBRARY NAME tuple_go PATH fdb/tuple)
 add_dependencies(tuple_go fdb_go)
