--- a/cmake/InstallLayout.cmake
+++ b/cmake/InstallLayout.cmake
@@ -99,7 +99,7 @@ install_destinations(TGZ
   BIN bin
-  SBIN sbin
+  SBIN bin
   LIB lib
-  FDBMONITOR sbin
+  FDBMONITOR bin
   BACKUPAGENT usr/lib/foundationdb
   INCLUDE include
   ETC etc/foundationdb
@@ -108,9 +108,9 @@ copy_install_destinations(TGZ VERSIONED
 copy_install_destinations(TGZ VERSIONED PREFIX "usr/lib/foundationdb-${FDB_VERSION}${FDB_BUILDTIME_STRING}/")
 install_destinations(DEB
   BIN usr/bin
-  SBIN usr/sbin
+  SBIN usr/bin
   LIB usr/lib
   FDBMONITOR usr/lib/foundationdb
   BACKUPAGENT usr/lib/foundationdb
   INCLUDE usr/include
   ETC etc/foundationdb
