diff --git a/bindings/c/test/apitester/TesterTestSpec.cpp b/bindings/c/test/apitester/TesterTestSpec.cpp
index fcf166775..b3f0a0d9f 100644
--- a/bindings/c/test/apitester/TesterTestSpec.cpp
+++ b/bindings/c/test/apitester/TesterTestSpec.cpp
@@ -125,10 +125,10 @@ std::string toml_to_string(const T& value) {
 	// and thus, is equivalent to testspec.  However, strings are quoted, so we
 	// must remove the quotes.
 	if (value.type() == toml::value_t::string) {
-		const std::string& formatted = toml::format(value);
+		const std::string& formatted = toml::format(value, toml::spec::v(1,0,0));
 		return formatted.substr(1, formatted.size() - 2);
 	} else {
-		return toml::format(value);
+		return toml::format(value, toml::spec::v(1,0,0));
 	}
 }
 
@@ -144,7 +144,7 @@ TestSpec readTomlTestSpec(std::string fileName) {
 	const toml::value& conf = toml::parse(fileName);
 
 	// Then parse each test
-	const toml::array& tests = toml::find(conf, "test").as_array();
+	const toml::array& tests = toml::get<toml::array>(toml::find(conf, "test"));
 	if (tests.size() == 0) {
 		throw TesterError("Invalid test file. No [test] section found");
 	} else if (tests.size() > 1) {
@@ -154,7 +154,7 @@ TestSpec readTomlTestSpec(std::string fileName) {
 	const toml::value& test = tests[0];
 
 	// First handle all test-level settings
-	for (const auto& [k, v] : test.as_table()) {
+	for (const auto& [k, v] : toml::get<toml::table>(test)) {
 		if (k == "workload") {
 			continue;
 		}
@@ -168,9 +168,9 @@ TestSpec readTomlTestSpec(std::string fileName) {
 
 	// Look for "knobs" section. If exists, store all options as knob key-value pairs
 	try {
-		const toml::array& overrideKnobs = toml::find(conf, "knobs").as_array();
+		const toml::array& overrideKnobs = toml::get<toml::array>(toml::find(conf, "knobs"));
 		for (const toml::value& knob : overrideKnobs) {
-			for (const auto& [key, value_] : knob.as_table()) {
+			for (const auto& [key, value_] : toml::get<toml::table>(knob)) {
 				const std::string& value = toml_to_string(value_);
 				spec.knobs.emplace_back(key, value);
 			}
@@ -180,11 +180,11 @@ TestSpec readTomlTestSpec(std::string fileName) {
 	}
 
 	// And then copy the workload attributes to spec.options
-	const toml::array& workloads = toml::find(test, "workload").as_array();
+	const toml::array& workloads = toml::get<toml::array>(toml::find(test, "workload"));
 	for (const toml::value& workload : workloads) {
 		workloadSpec = WorkloadSpec();
 		auto& options = workloadSpec.options;
-		for (const auto& [attrib, v] : workload.as_table()) {
+		for (const auto& [attrib, v] : toml::get<toml::table>(workload)) {
 			options[attrib] = toml_to_string(v);
 		}
 		auto itr = options.find("name");
diff --git a/fdbserver/SimulatedCluster.actor.cpp b/fdbserver/SimulatedCluster.actor.cpp
index c78a2cb90..7c1dfc9bf 100644
--- a/fdbserver/SimulatedCluster.actor.cpp
+++ b/fdbserver/SimulatedCluster.actor.cpp
@@ -116,7 +116,7 @@ bool destructed = false;
 // environment details
 class TestConfig : public BasicTestConfig {
 	class ConfigBuilder {
-		using value_type = toml::basic_value<toml::discard_comments>;
+		using value_type = toml::value;
 		using base_variant = std::
 		    variant<int, float, double, bool, std::string, std::vector<int>, std::vector<std::string>, ConfigDBType>;
 		using types =
@@ -126,20 +126,20 @@ class TestConfig : public BasicTestConfig {
 		struct visitor {
 			const value_type& value;
 			visitor(const value_type& v) : value(v) {}
-			void operator()(int* val) const { *val = value.as_integer(); }
-			void operator()(Optional<int>* val) const { *val = value.as_integer(); }
-			void operator()(float* val) const { *val = value.as_floating(); }
-			void operator()(Optional<float>* val) const { *val = value.as_floating(); }
-			void operator()(double* val) const { *val = value.as_floating(); }
-			void operator()(Optional<double>* val) const { *val = value.as_floating(); }
-			void operator()(bool* val) const { *val = value.as_boolean(); }
-			void operator()(Optional<bool>* val) const { *val = value.as_boolean(); }
-			void operator()(std::string* val) const { *val = value.as_string(); }
-			void operator()(Optional<std::string>* val) const { *val = value.as_string(); }
+			void operator()(int* val) const { *val = toml::get<int>(value); }
+			void operator()(Optional<int>* val) const { *val = toml::get<int>(value); }
+			void operator()(float* val) const { *val = toml::get<double>(value); }
+			void operator()(Optional<float>* val) const { *val = toml::get<double>(value); }
+			void operator()(double* val) const { *val = toml::get<double>(value); }
+			void operator()(Optional<double>* val) const { *val = toml::get<double>(value); }
+			void operator()(bool* val) const { *val = toml::get<bool>(value); }
+			void operator()(Optional<bool>* val) const { *val = toml::get<bool>(value); }
+			void operator()(std::string* val) const { *val = toml::get<std::string>(value); }
+			void operator()(Optional<std::string>* val) const { *val = toml::get<std::string>(value); }
 			void operator()(std::vector<int>* val) const {
-				auto arr = value.as_array();
+				auto arr = toml::get<toml::array>(value);
 				for (const auto& i : arr) {
-					val->emplace_back(i.as_integer());
+					val->emplace_back(toml::get<int>(i));
 				}
 			}
 			void operator()(Optional<std::vector<int>>* val) const {
@@ -148,10 +148,10 @@ class TestConfig : public BasicTestConfig {
 				*val = std::move(res);
 			}
 			void operator()(ConfigDBType* val) const {
-				if (value.as_string() == "random") {
+				if (toml::get<std::string>(value) == "random") {
 					*val = deterministicRandom()->coinflip() ? ConfigDBType::SIMPLE : ConfigDBType::PAXOS;
 				} else {
-					*val = configDBTypeFromString(value.as_string());
+					*val = configDBTypeFromString(toml::get<std::string>(value));
 				}
 			}
 			void operator()(Optional<ConfigDBType>* val) const {
@@ -160,9 +160,9 @@ class TestConfig : public BasicTestConfig {
 				*val = std::move(res);
 			}
 			void operator()(std::vector<std::string>* val) const {
-				auto arr = value.as_array();
+				auto arr = toml::get<toml::array>(value);
 				for (const auto& i : arr) {
-					val->emplace_back(i.as_string());
+					val->emplace_back(toml::get<std::string>(i));
 				}
 			}
 			void operator()(Optional<std::vector<std::string>>* val) const {
@@ -458,13 +458,13 @@ public:
 
 	bool tomlKeyPresent(const toml::value& data, std::string key) {
 		if (data.is_table()) {
-			for (const auto& [k, v] : data.as_table()) {
+			for (const auto& [k, v] : toml::get<toml::table>(data)) {
 				if (k == key || tomlKeyPresent(v, key)) {
 					return true;
 				}
 			}
 		} else if (data.is_array()) {
-			for (const auto& v : data.as_array()) {
+			for (const auto& v : toml::get<toml::array>(data)) {
 				if (tomlKeyPresent(v, key)) {
 					return true;
 				}
@@ -529,7 +529,7 @@ public:
 		try {
 			auto file = toml::parse(testFile);
 			if (file.contains("configuration") && toml::find(file, "configuration").is_table()) {
-				auto conf = toml::find(file, "configuration").as_table();
+				auto conf = toml::get<toml::table>(toml::find(file, "configuration"));
 				for (const auto& [key, value] : conf) {
 					if (key == "ClientInfoLogging") {
 						setNetworkOption(FDBNetworkOptions::DISABLE_CLIENT_STATISTICS_LOGGING);
diff --git a/fdbserver/tester.actor.cpp b/fdbserver/tester.actor.cpp
index 6e2027b38..6edd70af9 100644
--- a/fdbserver/tester.actor.cpp
+++ b/fdbserver/tester.actor.cpp
@@ -2333,10 +2333,10 @@ std::string toml_to_string(const T& value) {
 	// and thus, is equivalent to testspec.  However, strings are quoted, so we
 	// must remove the quotes.
 	if (value.type() == toml::value_t::string) {
-		const std::string& formatted = toml::format(value);
+		const std::string& formatted = toml::format(value, toml::spec::v(1,0,0));
 		return formatted.substr(1, formatted.size() - 2);
 	} else {
-		return toml::format(value);
+		return toml::format(value, toml::spec::v(1,0,0));
 	}
 }
 
@@ -2353,9 +2353,9 @@ KnobKeyValuePairs getOverriddenKnobKeyValues(const toml::value& context) {
 	KnobKeyValuePairs result;
 
 	try {
-		const toml::array& overrideKnobs = toml::find(context, "knobs").as_array();
+		const toml::array& overrideKnobs = toml::get<toml::array>(toml::find(context, "knobs"));
 		for (const toml::value& knob : overrideKnobs) {
-			for (const auto& [key, value_] : knob.as_table()) {
+			for (const auto& [key, value_] : toml::get<toml::table>(knob)) {
 				const std::string& value = toml_to_string(value_);
 				ParsedKnobValue parsedValue = CLIENT_KNOBS->parseKnobValue(key, value);
 				if (std::get_if<NoKnobFound>(&parsedValue)) {
@@ -2392,12 +2392,12 @@ TestSet readTOMLTests_(std::string fileName) {
 	result.overrideKnobs = getOverriddenKnobKeyValues(conf);
 
 	// Then parse each test
-	const toml::array& tests = toml::find(conf, "test").as_array();
+	const toml::array& tests = toml::get<toml::array>(toml::find(conf, "test"));
 	for (const toml::value& test : tests) {
 		TestSpec spec;
 
 		// First handle all test-level settings
-		for (const auto& [k, v] : test.as_table()) {
+		for (const auto& [k, v] : toml::get<toml::table>(test)) {
 			if (k == "workload" || k == "knobs") {
 				continue;
 			}
@@ -2411,11 +2411,11 @@ TestSet readTOMLTests_(std::string fileName) {
 		}
 
 		// And then copy the workload attributes to spec.options
-		const toml::array& workloads = toml::find(test, "workload").as_array();
+		const toml::array& workloads = toml::get<toml::array>(toml::find(test, "workload"));
 		for (const toml::value& workload : workloads) {
 			workloadOptions = Standalone<VectorRef<KeyValueRef>>();
 			TraceEvent("TestParserFlush").detail("Reason", "new (compound) test");
-			for (const auto& [attrib, v] : workload.as_table()) {
+			for (const auto& [attrib, v] : toml::get<toml::table>(workload)) {
 				const std::string& value = toml_to_string(v);
 				workloadOptions.push_back_deep(workloadOptions.arena(),
 				                               KeyValueRef(StringRef(attrib), StringRef(value)));

