diff --git a/src/dev/build/tasks/build_packages_task.ts b/src/dev/build/tasks/build_packages_task.ts
index 98ced278c..f022f00e7 100644
--- a/src/dev/build/tasks/build_packages_task.ts
+++ b/src/dev/build/tasks/build_packages_task.ts
@@ -308,7 +308,7 @@ export const BuildPackages: Task = {
 };
 
 export async function buildWebpackBundles({ quiet, dist }: { quiet: boolean; dist: boolean }) {
-  const options = [quiet ? ['--quiet'] : [], dist ? ['--dist'] : []].flat();
+  const options = [['--allow-root'], quiet ? ['--quiet'] : [], dist ? ['--dist'] : []].flat();
   const stdio: StdioOption[] = quiet
     ? ['ignore', 'pipe', 'pipe']
     : ['inherit', 'inherit', 'inherit'];
