diff --git a/systemd/jenkins.service b/systemd/jenkins.service
index 724dc1a..327d998 100644
--- a/systemd/jenkins.service
+++ b/systemd/jenkins.service
@@ -41,7 +41,7 @@ Environment="JENKINS_HOME=/var/lib/@@ARTIFACTNAME@@"
 WorkingDirectory=/var/lib/@@ARTIFACTNAME@@
 
 # Location of the Jenkins WAR
-#Environment="JENKINS_WAR=/usr/share/java/@@ARTIFACTNAME@@.war"
+Environment="JENKINS_WAR=/usr/share/java/jenkins/@@ARTIFACTNAME@@.war"
 
 # Location of the exploded WAR
 Environment="JENKINS_WEBROOT=%C/@@ARTIFACTNAME@@/war"
