diff --git a/systemd/jenkins.service b/systemd/jenkins.service
index 724dc1a..ac800be 100644
--- a/systemd/jenkins.service
+++ b/systemd/jenkins.service
@@ -22,6 +22,10 @@ NotifyAccess=main
 ExecStart=/usr/bin/@@ARTIFACTNAME@@
 Restart=on-failure
 SuccessExitStatus=143
+RuntimeDirectory=@@ARTIFACTNAME@@
+StateDirectory=@@ARTIFACTNAME@@
+CacheDirectory=@@ARTIFACTNAME@@
+LogsDirectory=@@ARTIFACTNAME@@
 
 # Configures the time to wait for start-up. If Jenkins does not signal start-up
 # completion within the configured time, the service will be considered failed
diff --git a/systemd/jenkins.sh b/systemd/jenkins.sh
index 3aadf77..931a032 100755
--- a/systemd/jenkins.sh
+++ b/systemd/jenkins.sh
@@ -144,7 +144,7 @@ else
 fi
 
 if [ -z "${JENKINS_WAR}" ]; then
-	JENKINS_WAR=/usr/share/java/@@ARTIFACTNAME@@.war
+	JENKINS_WAR=/usr/share/java/jenkins/@@ARTIFACTNAME@@.war
 fi
 
 check_env \
