--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -99,7 +99,7 @@ if (DCGM_BUILD_COVERAGE)
     set(DCGM_CXX_FLAGS "${DCGM_CXX_FLAGS} --coverage")
 endif ()
 
-set(DCGM_CXX_FLAGS "${DCGM_CXX_FLAGS} -fmax-errors=1 -Werror -fno-omit-frame-pointer")
+set(DCGM_CXX_FLAGS "${DCGM_CXX_FLAGS} -fmax-errors=1 -Wno-error -fno-omit-frame-pointer")
 
 # Set common CXXFLAGS as a base for CFLAGS
 set(DCGM_C_FLAGS ${DCGM_CXX_FLAGS})
