diff --git a/Gemfile b/Gemfile
index 22c41ae5763..37a1816a4cd 100644
--- a/Gemfile
+++ b/Gemfile
@@ -147,7 +147,7 @@ gem 'browser', '~> 5.3.1', feature_category: :shared
 gem 'ohai', '~> 18.1', feature_category: :product_analytics
 
 # GPG
-gem 'gpgme', '~> 2.0.24', feature_category: :source_code_management
+gem 'gpgme', '~> 2.0.25', feature_category: :source_code_management
 
 # LDAP Auth
 # GitLab fork with several improvements to original library. For full list of changes
diff --git a/Gemfile.lock b/Gemfile.lock
index b88058fb013..3afd5d2910f 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -2202,7 +2202,7 @@ DEPENDENCIES
   google-cloud-storage (~> 1.45.0)
   google-protobuf (~> 3.25, >= 3.25.3)
   googleauth (~> 1.14)
-  gpgme (~> 2.0.24)
+  gpgme (~> 2.0.25)
   grape (~> 2.0.0)
   grape-entity (~> 1.0.1)
   grape-path-helpers (~> 2.0.1)
