diff --git a/bitnami/external-dns/templates/_helpers.tpl b/bitnami/external-dns/templates/_helpers.tpl
index 27bba581be..d9d24b3105 100644
--- a/bitnami/external-dns/templates/_helpers.tpl
+++ b/bitnami/external-dns/templates/_helpers.tpl
@@ -365,14 +365,6 @@ external-dns: pdns.apiKey
 {{- end -}}
 {{- end -}}
 
-{{/* Check if there are rolling tags in the images */}}
-{{- define "external-dns.checkRollingTags" -}}
-{{- if and (contains "bitnami/" .Values.image.repository) (not (.Values.image.tag | toString | regexFind "-r\\d+$|sha256:")) }}
-WARNING: Rolling tag detected ({{ .Values.image.repository }}:{{ .Values.image.tag }}), please note that it is strongly recommended to avoid using rolling tags in a production environment.
-+info https://techdocs.broadcom.com/us/en/vmware-tanzu/application-catalog/tanzu-application-catalog/services/tac-doc/apps-tutorials-understand-rolling-tags-containers-index.html
-{{- end }}
-{{- end -}}
-
 {{/*
 Validate values of Azure DNS:
 - must provide the Azure Resource Group when provider is "azure" and tenantId is set
