diff --git a/bitnami/external-dns/templates/_helpers.tpl b/bitnami/external-dns/templates/_helpers.tpl
index 27bba581be..d9d24b3105 100644
--- a/bitnami/external-dns/templates/_helpers.tpl
+++ b/bitnami/external-dns/templates/_helpers.tpl
@@ -365,11 +365,6 @@ external-dns: pdns.apiKey
 {{- end -}}
 {{- end -}}
 
-{{/* Check if there are rolling tags in the images */}}
-{{- define "external-dns.checkRollingTags" -}}
-{{- include "common.warnings.rollingTag" .Values.image }}
-{{- end -}}
-
 {{/*
 Validate values of Azure DNS:
 - must provide the Azure Resource Group when provider is "azure" and tenantId is set
