From 7c96307342346a63e164c24702e384d4e6a11f1a Mon Sep 17 00:00:00 2001
From: Markus Boehme <markus.boehme@chainguard.dev>
Date: Fri, 7 Nov 2025 11:32:37 +0100
Subject: [PATCH] systemd.te: allow timesyncd to use NTP to sync clocks

Signed-off-by: Markus Boehme <markus.boehme@chainguard.dev>
---
 policy/modules/system/systemd.te | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index b70e415bd..4a4a401ca 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -1195,6 +1195,11 @@ corecmd_dontaudit_access_check_bin(systemd_timedated_t)
 
 corenet_tcp_connect_time_port(systemd_timedated_t)
 
+corenet_udp_bind_all_nodes(systemd_timedated_t)
+corenet_udp_bind_ntp_port(systemd_timedated_t)
+corenet_udp_sendrecv_ntp_port(systemd_timedated_t)
+corenet_sendrecv_ntp_client_packets(systemd_timedated_t)
+
 dev_rw_realtime_clock(systemd_timedated_t)
 dev_write_kmsg(systemd_timedated_t)
 dev_read_sysfs(systemd_timedated_t)
-- 
2.51.0

