diff --git a/pkg/aws/pca.go b/pkg/aws/pca.go
index 22a77b73..5299d2e8 100644
--- a/pkg/aws/pca.go
+++ b/pkg/aws/pca.go
@@ -19,7 +19,7 @@ package aws
 import (
 	"bytes"
 	"context"
-	"crypto/md5"
+	"crypto/sha256"
 	"encoding/pem"
 	"fmt"
 	"strings"
@@ -92,7 +92,7 @@ func NewProvisioner(config aws.Config, arn string) (p *PCAProvisioner) {
 // @see: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/Run_Instance_Idempotency.html
 func idempotencyToken(cr *cmapi.CertificateRequest) string {
 	token := []byte(cr.ObjectMeta.Namespace + "/" + cr.ObjectMeta.Name)
-	return fmt.Sprintf("%x", md5.Sum(token))
+	return fmt.Sprintf("%x", sha256.Sum256(token))
 }
 
 // Sign takes a certificate request and signs it using PCA
diff --git a/pkg/aws/pca_test.go b/pkg/aws/pca_test.go
index d81e0c79..dcf4a66d 100644
--- a/pkg/aws/pca_test.go
+++ b/pkg/aws/pca_test.go
@@ -309,7 +309,7 @@ func TestPCATemplateArn(t *testing.T) {
 
 func TestIdempotencyToken(t *testing.T) {
 	var (
-		idempotencyTokenMaxLength = 36
+		idempotencyTokenMaxLength = 64 // Updated max length
 	)
 
 	type testCase struct {
@@ -325,7 +325,7 @@ func TestIdempotencyToken(t *testing.T) {
 					Namespace: "fake-namespace",
 				},
 			},
-			expected: "f331cbfd0cc6569f58c12c3dbb238a4f",
+			expected: "6c414adffafef3d366beabcfcd74e8fa5a724beb4d3aa61e9e6ccdf32693b2e1", // Updated SHA-256 hash
 		},
 	}