diff --git a/test/hotspot/gtest/gc/shenandoah/test_shenandoahNumberSeq.cpp b/test/hotspot/gtest/gc/shenandoah/test_shenandoahNumberSeq.cpp
--- a/test/hotspot/gtest/gc/shenandoah/test_shenandoahNumberSeq.cpp
+++ b/test/hotspot/gtest/gc/shenandoah/test_shenandoahNumberSeq.cpp
@@ -158,6 +158,6 @@ TEST_VM_F(ShenandoahNumberSeqMergeTest, merge_test) {
   EXPECT_NEAR(merged.sd(),  seq3.sd(),  err);
 
   // These are not implemented
-  EXPECT_TRUE(isnan(merged.davg()));
-  EXPECT_TRUE(isnan(merged.dvariance()));
+  EXPECT_TRUE(std::isnan(merged.davg()));
+  EXPECT_TRUE(std::isnan(merged.dvariance()));
 }
