From fbd02674853bf8d1d59edfe8c4f3941ff078dbb0 Mon Sep 17 00:00:00 2001
From: Amelia Crate <amelia.crate@chainguard.dev>
Date: Tue, 24 Jun 2025 08:46:21 -0700
Subject: [PATCH] crypto: log build-time kernel version in jent init

---
 crypto/jitterentropy.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/crypto/jitterentropy.c b/crypto/jitterentropy.c
index 3b390bd6c..b86afce30 100644
--- a/crypto/jitterentropy.c
+++ b/crypto/jitterentropy.c
@@ -145,6 +145,8 @@ struct rand_data {
  */
 #define JENT_ENTROPY_SAFETY_FACTOR	64
 
+#include <linux/printk.h>
+#include <generated/utsrelease.h>
 #include <linux/fips.h>
 #include <linux/minmax.h>
 #include "jitterentropy.h"
@@ -707,6 +709,7 @@ void jent_entropy_collector_free(struct rand_data *entropy_collector)
 int jent_entropy_init(unsigned int osr, unsigned int flags, void *hash_state,
 		      struct rand_data *p_ec)
 {
+	printk(KERN_INFO "Initializing jitterentropy built from %s sources\n", UTS_RELEASE);
 	/*
 	 * If caller provides an allocated ec, reuse it which implies that the
 	 * health test entropy data is used to further still the available
-- 
2.43.0

