From 0f973b1e7b1bb0c099f88e343267be52782f3717 Mon Sep 17 00:00:00 2001
From: Julia Volmer <julia.volmer@arangodb.com>
Date: Mon, 23 Jun 2025 14:39:57 +0200
Subject: [PATCH] Fix load memory order

Origin: upstream, https://github.com/arangodb/arangodb/pull/21818
Bug: https://github.com/arangodb/arangodb/issues/21811
Bug-Chainguard: https://github.com/chainguard-dev/internal-dev/issues/14079

---
 lib/Containers/Concurrent/ThreadOwnedList.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/Containers/Concurrent/ThreadOwnedList.h b/lib/Containers/Concurrent/ThreadOwnedList.h
index 95df13e18b05..2ea1a06d528b 100644
--- a/lib/Containers/Concurrent/ThreadOwnedList.h
+++ b/lib/Containers/Concurrent/ThreadOwnedList.h
@@ -236,7 +236,7 @@ struct ThreadOwnedList
       current = next;
       next = next->next_to_free;
       // (9) - this load synchronizes with the store in (6) and (8)
-      if (current->previous.load(std::memory_order_release) != nullptr) {
+      if (current->previous.load(std::memory_order_acquire) != nullptr) {
         if (_metrics) {
           _metrics->decrement_ready_for_deletion_nodes();
         }
