diff --git a/LICENSE-binary b/LICENSE-binary
index 758a230d15..3dc1161d65 100644
--- a/LICENSE-binary
+++ b/LICENSE-binary
@@ -207,12 +207,12 @@ License Version 2.0:
 
 audience-annotations-0.12.0
 caffeine-2.9.3
-commons-beanutils-1.9.4
+commons-beanutils-1.11.0
 commons-cli-1.4
 commons-collections-3.2.2
 commons-digester-2.1
 commons-io-2.14.0
-commons-lang3-3.12.0
+commons-lang3-3.18.0
 commons-logging-1.2
 commons-validator-1.7
 error_prone_annotations-2.10.0
diff --git a/build.gradle b/build.gradle
index 6b4725d542..426311420f 100644
--- a/build.gradle
+++ b/build.gradle
@@ -151,6 +151,9 @@ allprojects {
     if (name != "zinc") {
       resolutionStrategy {
         force(
+          // Enforce the updated commons-beanutils to address CVE
+          libs.commonsBeanutils,
+          libs.commonsLang3,
           // be explicit about the javassist dependency version instead of relying on the transitive version
           libs.javassist,
           // ensure we have a single version in the classpath despite transitive dependencies
@@ -935,6 +938,8 @@ project(':core') {
     implementation project(':server')
 
     implementation libs.argparse4j
+    implementation libs.commonsBeanutils // Explicitly adding commons-beanutils to override transitive dependency
+    implementation libs.commonsLang3
     implementation libs.commonsValidator
     implementation libs.jacksonDatabind
     implementation libs.jacksonModuleScala
diff --git a/gradle/dependencies.gradle b/gradle/dependencies.gradle
index 4e6f83fade..1d55afe908 100644
--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -89,8 +89,10 @@ versions += [
   // CVE-2023-2976 and CVE-2020-8908 can be dropped from
   // gradle/resources/dependencycheck-suppressions.xml
   checkstyle: "8.36.2",
+  commonsBeanutils: "1.11.0", // Added to address CVE GHSA-wxr5-93ph-8wr9
   commonsCli: "1.4",
   commonsIo: "2.14.0", // ZooKeeper dependency. Do not use, this is going away.
+  commonsLang3: "3.18.0",
   commonsValidator: "1.7",
   dropwizardMetrics: "4.1.12.1",
   gradle: "8.10.2",
@@ -181,7 +183,9 @@ libs += [
   bcpkix: "org.bouncycastle:bcpkix-jdk18on:$versions.bcpkix",
   caffeine: "com.github.ben-manes.caffeine:caffeine:$versions.caffeine",
   commonsCli: "commons-cli:commons-cli:$versions.commonsCli",
+  commonsBeanutils: "commons-beanutils:commons-beanutils:$versions.commonsBeanutils",
   commonsIo: "commons-io:commons-io:$versions.commonsIo",
+  commonsLang3: "org.apache.commons:commons-lang3:$versions.commonsLang3",
   commonsValidator: "commons-validator:commons-validator:$versions.commonsValidator",
   jacksonAnnotations: "com.fasterxml.jackson.core:jackson-annotations:$versions.jackson",
   jacksonDatabind: "com.fasterxml.jackson.core:jackson-databind:$versions.jackson",
