From 10a22616099b243ba8d372df0fd352ae434728fc Mon Sep 17 00:00:00 2001
From: Ankush Pathak <ankush.pathak@chainguard.dev>
Date: Wed, 19 Nov 2025 19:36:09 +0000
Subject: [PATCH] Bump jersey to version 2.46 and remediate GHSA-7p63-w6x9-6gr7

Signed-off-by: Ankush Pathak <ankush.pathak@chainguard.dev>
---
 LICENSE-binary             | 12 ++++++------
 gradle/dependencies.gradle |  2 +-
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/LICENSE-binary b/LICENSE-binary
index 758a230d15..2eaec3ccd1 100644
--- a/LICENSE-binary
+++ b/LICENSE-binary
@@ -297,12 +297,12 @@ hk2-utils-2.6.1
 osgi-resource-locator-1.0.3
 aopalliance-repackaged-2.6.1
 jakarta.inject-2.6.1
-jersey-client-2.39.1
-jersey-common-2.39.1
-jersey-container-servlet-2.39.1
-jersey-container-servlet-core-2.39.1
-jersey-hk2-2.39.1
-jersey-server-2.39.1
+jersey-client-2.46
+jersey-common-2.46
+jersey-container-servlet-2.46
+jersey-container-servlet-core-2.46
+jersey-hk2-2.46
+jersey-server-2.46
 
 ---------------------------------------
 CDDL 1.1 + GPLv2 with classpath exception
diff --git a/gradle/dependencies.gradle b/gradle/dependencies.gradle
index 4e6f83fade..fc8f53e190 100644
--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -100,7 +100,7 @@ versions += [
   jacoco: "0.8.10",
   javassist: "3.29.2-GA",
   jetty: "9.4.57.v20241219",
-  jersey: "2.39.1",
+  jersey: "2.46",
   jline: "3.25.1",
   jmh: "1.37",
   hamcrest: "2.2",
-- 
2.51.2

