From ca13e41796b9790924f3e7798136004a9884d838 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Mon, 2 Sep 2024 16:39:52 +0100
Subject: [PATCH 02/11] Always enable CGO for tools, for FIPS

---
 src/cmd/dist/build.go | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/src/cmd/dist/build.go b/src/cmd/dist/build.go
index 85512e3822..5dedec8802 100644
--- a/src/cmd/dist/build.go
+++ b/src/cmd/dist/build.go
@@ -1383,13 +1383,6 @@ func timelog(op, name string) {
 // to switch between the host and target configurations when cross-compiling.
 func toolenv() []string {
 	var env []string
-	if !mustLinkExternal(goos, goarch, false) {
-		// Unless the platform requires external linking,
-		// we disable cgo to get static binaries for cmd/go and cmd/pprof,
-		// so that they work on systems without the same dynamic libraries
-		// as the original build system.
-		env = append(env, "CGO_ENABLED=0")
-	}
 	if isRelease || os.Getenv("GO_BUILDER_NAME") != "" {
 		// Add -trimpath for reproducible builds of releases.
 		// Include builders so that -trimpath is well-tested ahead of releases.
-- 
2.51.0

