From 982b38beb876c5255093a5ac5acd7eac9e02d68d Mon Sep 17 00:00:00 2001
From: Uti Edun <uti.edun@chainguard.dev>
Date: Sat, 4 Oct 2025 19:11:15 +0000
Subject: [PATCH] Adding patch for int conversion to bool

---
 src/pybind/rgw/rgw.pyx | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/pybind/rgw/rgw.pyx b/src/pybind/rgw/rgw.pyx
index 9bbcdfff586..d210a70bbb8 100644
--- a/src/pybind/rgw/rgw.pyx
+++ b/src/pybind/rgw/rgw.pyx
@@ -7,6 +7,7 @@ from cpython cimport PyObject, ref, exc, array
 from libc.stdint cimport *
 from libc.stdlib cimport malloc, realloc, free
 from cstat cimport stat
+cimport libcpp
 
 IF BUILD_DOC:
     include "mock_rgw.pxi"
@@ -373,7 +374,7 @@ cdef class LibRGWFS(object):
         cdef:
             rgw_file_handle *_dir_handler = <rgw_file_handle*>dir_handler.handler
             uint64_t _offset = offset
-            bint _eof
+            libcpp.bool _eof
             uint32_t _flags = flags
         with nogil:
             ret = rgw_readdir(self.fs, _dir_handler, &_offset, &readdir_cb,
-- 
2.51.0

