Provide missing namespaces
--- a/common/DcgmStringHelpers.cpp
+++ b/common/DcgmStringHelpers.cpp
@@ -15,6 +15,7 @@
  */
 #include "DcgmStringHelpers.h"
 
+#include <algorithm>
 #include <cctype>
 #include <cstring>
 #include <ranges>

--- a/nvvs/include/IgnoreErrorCodesHelper.h
+++ b/nvvs/include/IgnoreErrorCodesHelper.h
@@ -20,6 +20,7 @@
 #include "DcgmStringHelpers.h"
 
 #include <exception>
+#include <optional>
 #include <map>
 #include <string>
 #include <unordered_set>

