diff --git a/third_party/nvidia/lib/NVGPUToLLVM/CMakeLists.txt b/third_party/nvidia/lib/NVGPUToLLVM/CMakeLists.txt
index 9d7ae73..bc5eee1 100644
--- a/third_party/nvidia/lib/NVGPUToLLVM/CMakeLists.txt
+++ b/third_party/nvidia/lib/NVGPUToLLVM/CMakeLists.txt
@@ -3,6 +3,7 @@ add_triton_library(NVGPUToLLVM
 
     DEPENDS
     NVGPUConversionPassIncGen
+    TritonTableGen
 
     LINK_LIBS PUBLIC
     NVGPUIR
