diff --git a/build.gradle b/build.gradle
index 6b4725d542..426311420f 100644
--- a/build.gradle
+++ b/build.gradle
@@ -151,6 +151,9 @@ allprojects {
     if (name != "zinc") {
       resolutionStrategy {
         force(
+          // Enforce the updated commons-beanutils to address CVE
+          libs.commonsBeanutils,
+          libs.commonsLang3,
           // be explicit about the javassist dependency version instead of relying on the transitive version
           libs.javassist,
           // ensure we have a single version in the classpath despite transitive dependencies
@@ -935,6 +938,8 @@ project(':core') {
     implementation project(':server')
 
     implementation libs.argparse4j
+    implementation libs.commonsBeanutils // Explicitly adding commons-beanutils to override transitive dependency
+    implementation libs.commonsLang3
     implementation libs.commonsValidator
     implementation libs.jacksonDatabind
     implementation libs.jacksonModuleScala
diff --git a/gradle/dependencies.gradle b/gradle/dependencies.gradle
index f2e80cc..071e7c9 100644
--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -89,9 +89,11 @@ versions += [
   // CVE-2023-2976 and CVE-2020-8908 can be dropped from
   // gradle/resources/dependencycheck-suppressions.xml
   checkstyle: "8.36.2",
+  commonsBeanutils: "1.11.0", // Added to address CVE GHSA-wxr5-93ph-8wr9
   commonsCli: "1.4",
   commonsIo: "2.14.0", // ZooKeeper dependency. Do not use, this is going away.
   commonsLang: "3.18.0",
+  commonsLang3: "3.18.0",
   commonsValidator: "1.10.1",
   dropwizardMetrics: "4.1.12.1",
   gradle: "8.10.2",
@@ -184,8 +186,10 @@ libs += [
   bcpkix: "org.bouncycastle:bcpkix-jdk18on:$versions.bcpkix",
   caffeine: "com.github.ben-manes.caffeine:caffeine:$versions.caffeine",
   commonsCli: "commons-cli:commons-cli:$versions.commonsCli",
+  commonsBeanutils: "commons-beanutils:commons-beanutils:$versions.commonsBeanutils",
   commonsIo: "commons-io:commons-io:$versions.commonsIo",
   commonsLang: "org.apache.commons:commons-lang3:$versions.commonsLang",
+  commonsLang3: "org.apache.commons:commons-lang3:$versions.commonsLang3",
   commonsValidator: "commons-validator:commons-validator:$versions.commonsValidator",
   jacksonAnnotations: "com.fasterxml.jackson.core:jackson-annotations:$versions.jackson",
   jacksonDatabind: "com.fasterxml.jackson.core:jackson-databind:$versions.jackson",
