From b7005be8693de322569c00779fde0f70ebf91fcc Mon Sep 17 00:00:00 2001
From: Markus Boehme <markus.boehme@chainguard.dev>
Date: Wed, 19 Nov 2025 14:52:55 -0500
Subject: [PATCH] ssh: allow ssh-agent forwarding

Signed-off-by: Markus Boehme <markus.boehme@chainguard.dev>
---
 policy/modules/services/ssh.te | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/policy/modules/services/ssh.te b/policy/modules/services/ssh.te
index e982492d2..913eb5909 100644
--- a/policy/modules/services/ssh.te
+++ b/policy/modules/services/ssh.te
@@ -477,6 +477,7 @@ corenet_sendrecv_xserver_server_packets(sshd_t)
 auth_exec_login_program(sshd_t)
 auth_signal_chk_passwd(sshd_t)
 
+userdom_create_user_home_dirs(sshd_t)
 userdom_read_user_home_content_files(sshd_t)
 userdom_read_user_home_content_symlinks(sshd_t)
 #userdom_manage_tmp_role(system_r, sshd_t)
@@ -485,6 +486,9 @@ userdom_signal_unpriv_users(sshd_t)
 userdom_dyntransition_unpriv_users(sshd_t)
 userdom_map_tmp_files(sshd_t)
 
+# for agent forwarding
+allow sshd_t user_home_dir_t:sock_file { create };
+
 tunable_policy(`ssh_sysadm_login',`
 	# Relabel and access ptys created by sshd
 	# ioctl is necessary for logout() processing for utmp entry and for w to
-- 
2.51.2

