diff --git a/scripts/docker-entrypoint.sh b/scripts/docker-entrypoint.sh
index d9d7d35..7a1da03 100755
--- a/scripts/docker-entrypoint.sh
+++ b/scripts/docker-entrypoint.sh
@@ -1,6 +1,11 @@
 #!/bin/bash
 set -e
 
+[[ "${C3_TRACE:-false}" == "true" ]] && set -o xtrace
+
+. /c3/bin/libcassandra.sh
+
+
 # first arg is `-f` or `--some-option`
 # or there are no args
 if [ "$#" -eq 0 ] || [ "${1#-}" != "$1" ]; then
@@ -8,12 +13,12 @@ if [ "$#" -eq 0 ] || [ "${1#-}" != "$1" ]; then
 fi
 
 if [ "$CASSANDRA_CONF" == "" ]; then
-    export CASSANDRA_CONF=/etc/cassandra
+    export CASSANDRA_CONF=/c3/cassandra/conf
 fi
 
 # allow the container to be started with `--user`
 if [ "$1" = 'mgmtapi' -a "$(id -u)" = '0' ]; then
-    find "$CASSANDRA_CONF" /var/lib/cassandra /var/log/cassandra \
+    find "$CASSANDRA_CONF" /c3/cassandra \
         \! -user cassandra -exec chown cassandra '{}' +
     # exec gosu cassandra "$BASH_SOURCE" "$@"
 fi
@@ -41,8 +46,23 @@ _sed-in-place() {
 }
 
 _metrics_collector_supported() {
-    # currently, metrics collector does not work on arm64
-    [ "$(uname -m)" != "aarch64" ] && [ -z "$MGMT_API_DISABLE_MCAC" ]
+    # Metrics collector shoud be enabled by default.
+    # Therefore only disable it if METRICS_COLLECTOR_DISABLED is set to true.
+    if [[ "$METRICS_COLLECTOR_DISABLED" == "true" ]]; then
+        echo "Metrics collector disabled from configuration."
+        return 1
+    fi
+
+    # currently, metrics collector does not work on arm
+    
+    if [[ "$(uname -m)" != "x86_64" ]]; then
+        echo "Metrics collector not supported because arch is not x86_64."
+        return 1
+    fi
+
+    echo "Metrics collector enabled."
+
+    return 0
 }
 
 _needs_nodetool_fix() {
@@ -68,11 +88,31 @@ EOF
 
 if [ "$1" = 'mgmtapi' ]; then
     echo "Starting Management API"
+    eval "$(cassandra_env)"
+    cassandra_clean_from_restart
+    cp -pr /opt/cassandra/* /c3/cassandra/
+    cassandra_copy_mounted_config
+    cassandra_copy_default_config
+    cassandra_setup_data_dirs
+
+
+    # TODO : handle read only file system
+    # https://c3energy.atlassian.net/browse/OP-31528
+    # https://c3energy.atlassian.net/browse/OP-32111
 
     # Copy over any config files mounted at /config
-    # cp /config/cassandra.yaml /etc/cassandra/cassandra.yaml
+    # cp /config/cassandra.yaml /c3/cassandra/conf/cassandra.yaml
     if [ -d "/config" ] && ! [ "/config" -ef "$CASSANDRA_CONF" ]; then
-        cp -R /config/* "${CASSANDRA_CONF:-/etc/cassandra}"
+        cp -R /config/* "${CASSANDRA_CONF:-/c3/cassandra/conf}"
+    fi
+
+    if [[ "$LOCAL_JMX" == "no"  ]]; then
+        if [[ -f $CASSANDRA_CONF/jmxremote.password ]]; then
+            # Change jmxremote.password to readonly permission
+            chmod 400 $CASSANDRA_CONF/jmxremote.password
+        else
+            echon "Warning : JMX is enabled but file ${CASSANDRA_CONF}/jmxremote.password is missing"
+        fi
     fi
 
     # Add the nodetool fix if needed
@@ -88,7 +128,7 @@ if [ "$1" = 'mgmtapi' ]; then
     #
     # MCAC metric filters are expected as an env variable in the following format:
     # "deny:org.apache.cassandra.metrics.table allow:org.apache.cassandra.metrics.table.test"
-    
+
     if _metrics_collector_supported && ! grep -qxF "JVM_OPTS=\"\$JVM_OPTS -javaagent:${MCAC_PATH}/lib/datastax-mcac-agent.jar\"" < ${CASSANDRA_CONF}/cassandra-env.sh ; then
         # ensure newline at end of file
         echo "" >> ${CASSANDRA_CONF}/cassandra-env.sh
