diff --git a/gradle/libs.versions.toml b/gradle/libs.versions.toml
index a4835c9f29c..65eb52f047a 100644
--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -13,7 +13,7 @@ jackson_databind  = "2.18.2"
 snakeyaml         = "2.1"
 icu4j             = "77.1"
 supercsv          = "2.4.0"
-log4j             = "2.21.0"
+log4j             = "2.25.3"
 slf4j             = "2.0.17"
 asm               = "9.7"
 jettison          = "1.5.4"
