diff --git a/distribution/src/bin/opensearch-env b/distribution/src/bin/opensearch-env
index 371b296a492..eefdf5891ca 100644
--- a/distribution/src/bin/opensearch-env
+++ b/distribution/src/bin/opensearch-env
@@ -44,7 +44,7 @@ if [[ -z "$OPENSEARCH_HOME" ]]; then
 fi
 
 # now set the classpath
-OPENSEARCH_CLASSPATH="$OPENSEARCH_HOME/lib/*"
+OPENSEARCH_CLASSPATH="$OPENSEARCH_HOME/lib/*:/usr/share/java/bouncycastle-fips/*"
 
 # now set the path to java: OPENSEARCH_JAVA_HOME -> JAVA_HOME -> bundled JRE -> bundled JDK
 if [ ! -z "$OPENSEARCH_JAVA_HOME" ]; then
@@ -114,7 +114,7 @@ OPENSEARCH_PATH_CONF=`cd "$OPENSEARCH_PATH_CONF"; pwd`
 
 # Check if any bc-fips jar exists on classpath
 # run in FIPS JVM if jar is found
-if ls "$OPENSEARCH_HOME/lib" | grep -E -q "bc-fips.*\.jar"; then
+if ls "/usr/share/java/bouncycastle-fips" | grep -E -q "bc-fips.*\.jar"; then
   echo "BouncyCastle FIPS library found, setting FIPS JVM options."
   export OPENSEARCH_JAVA_OPTS="-Dorg.bouncycastle.fips.approved_only=true \
     -Djava.security.properties=${OPENSEARCH_PATH_CONF}/fips_java.security \
