From cbb6adb2cec121dccf968cf73440cd0ee96e8c60 Mon Sep 17 00:00:00 2001
From: dann frazier <dann.frazier@chainguard.dev>
Date: Mon, 14 Jul 2025 22:49:25 +0000
Subject: [PATCH] Respect CUDA_HOME from the environment

Signed-off-by: dann frazier <dann.frazier@chainguard.dev>
---
 setup.py |   10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 3949ea8..b93b380 100644
--- a/setup.py
+++ b/setup.py
@@ -74,10 +74,16 @@ if enable_aot:
     from packaging.version import Version
 
     def get_cuda_version() -> Version:
-        if torch_cpp_ext.CUDA_HOME is None:
+        cuda_home = None
+        if 'CUDA_HOME' in os.environ:
+            cuda_home = os.environ['CUDA_HOME']
+        elif torch_cpp_ext.CUDA_HOME is None:
+            cuda_home = torch_cpp_ext.CUDA_HOME
+        if cuda_home is None:
             nvcc = "nvcc"
         else:
-            nvcc = os.path.join(torch_cpp_ext.CUDA_HOME, "bin/nvcc")
+            nvcc = os.path.join(cuda_home, "bin/nvcc")
+
         txt = subprocess.check_output([nvcc, "--version"], text=True)
         return Version(re.findall(r"release (\d+\.\d+),", txt)[0])
 
