diff --git a/build/Settings.cs b/build/Settings.cs
index 0b83627..f1d8a3d 100644
--- a/build/Settings.cs
+++ b/build/Settings.cs
@@ -112,6 +112,14 @@ namespace Build
                 RuntimeIdentifier = "linux-x64",
                 PublishReadyToRun = true,
                 PublishBinDirectorySubPath = Path.Combine("bin_v3", "linux-x64")
+            },
+            new BuildConfiguration()
+            {
+                ConfigId = ConfigId.any_any,
+                SourceProjectFileName = "extensions.csproj",
+                RuntimeIdentifier = "any",
+                PublishReadyToRun = false,
+                PublishBinDirectorySubPath = "bin"
             }
         };
 
@@ -162,4 +170,4 @@ namespace Build
             CsProjFilePath = Path.Combine(RootBuildDirectory, ConfigId.any_any.ToString(), "extensions.csproj")
         };
     }
-}
\ No newline at end of file
+}
