diff --git a/src/mongo/db/exec/plan_stats.h b/src/mongo/db/exec/plan_stats.h
index 30e066fe..e04678b2 100644
--- a/src/mongo/db/exec/plan_stats.h
+++ b/src/mongo/db/exec/plan_stats.h
@@ -31,6 +31,7 @@

 #include <cstdint>
 #include <cstdlib>
+#include <optional>
 #include <string>
 #include <vector>

diff --git a/src/mongo/db/query/plan_cache.cpp b/src/mongo/db/query/plan_cache.cpp
index cbeae376..72b8525f 100644
--- a/src/mongo/db/query/plan_cache.cpp
+++ b/src/mongo/db/query/plan_cache.cpp
@@ -38,6 +38,7 @@
 #include <algorithm>
 #include <math.h>
 #include <memory>
+#include <optional>
 #include <vector>

 #include "mongo/base/owned_pointer_vector.h"
