From a4c2204d11b867b5065e3e35eb71cedf77fe95ef Mon Sep 17 00:00:00 2001
From: Ajay K <ajaykemparaj@gmail.com>
Date: Thu, 6 Mar 2025 04:31:06 +0000
Subject: [PATCH] set gofips as latest always

---
 src/cmd/dist/build.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cmd/dist/build.go b/src/cmd/dist/build.go
index 8506e7383a..82e6e77ccc 100644
--- a/src/cmd/dist/build.go
+++ b/src/cmd/dist/build.go
@@ -179,7 +179,7 @@ func xinit() {
 
 	b = os.Getenv("GOFIPS140")
 	if b == "" {
-		b = "off"
+		b = "latest"
 	}
 	gofips140 = b
 
-- 
2.48.1

