diff --git a/scripts/build_wheel.py b/scripts/build_wheel.py
index dda7da6d0..0438ee28c 100755
--- a/scripts/build_wheel.py
+++ b/scripts/build_wheel.py
@@ -109,7 +109,7 @@ def create_venv(project_dir: Path):
 
     # Ensure compatible virtualenv version is installed (>=20.29.1, <22.0)
     print("-- Ensuring virtualenv version >=20.29.1,<22.0 is installed...")
-    build_run(f'"{sys.executable}" -m pip install "virtualenv>=20.29.1,<22.0"')
+    # build_run(f'"{sys.executable}" -m pip install "virtualenv>=20.29.1,<22.0"')
 
     # Create venv if it doesn't exist
     if not venv_prefix.exists():
@@ -197,7 +197,7 @@ def setup_venv(project_dir: Path, requirements_file: Path,
     print(
         f"-- Installing requirements from {requirements_file} into {venv_prefix}..."
     )
-    build_run(f'"{venv_python}" -m pip install -r "{requirements_file}"')
+    # build_run(f'"{venv_python}" -m pip install -r "{requirements_file}"')
 
     venv_conan = setup_conan(scripts_dir, venv_python)
 
@@ -205,7 +205,7 @@ def setup_venv(project_dir: Path, requirements_file: Path,
 
 
 def setup_conan(scripts_dir, venv_python):
-    build_run(f'"{venv_python}" -m pip install conan==2.14.0')
+    # build_run(f'"{venv_python}" -m pip install conan==2.14.0')
     # Determine the path to the conan executable within the venv
     venv_conan = scripts_dir / "conan"
     if not venv_conan.exists():
@@ -369,7 +369,7 @@ def generate_python_stubs_linux(binding_type: str, venv_python: Path,
     is_nanobind = binding_type == "nanobind"
     if is_nanobind:
         build_run(f"\"{venv_python}\" -m pip install nanobind")
-    build_run(f"\"{venv_python}\" -m pip install pybind11-stubgen")
+    # build_run(f"\"{venv_python}\" -m pip install pybind11-stubgen")
 
     env_stub_gen = os.environ.copy()
     cuda_home_dir = env_stub_gen.get("CUDA_HOME") or env_stub_gen.get(
@@ -417,7 +417,7 @@ def generate_python_stubs_windows(binding_type: str, venv_python: Path,
         print("Windows not yet supported for nanobind stubs")
         exit(1)
     else:
-        build_run(f"\"{venv_python}\" -m pip install pybind11-stubgen")
+        # build_run(f"\"{venv_python}\" -m pip install pybind11-stubgen")
         stubgen = "stubgen.py"
         stubgen_contents = """
                         # Loading torch, trt before bindings is required to avoid import errors on windows.

