diff --git a/bitnami/apisix/templates/_helpers.tpl b/bitnami/apisix/templates/_helpers.tpl
index 5b90f5cab9..7be9634257 100644
--- a/bitnami/apisix/templates/_helpers.tpl
+++ b/bitnami/apisix/templates/_helpers.tpl
@@ -251,11 +251,6 @@ Init container definition for waiting for the database to be ready
     - -ec
     - |
       #!/bin/bash
-      cp -R /opt/bitnami/apisix/conf /usr/local/apisix
-      ln -sf /opt/bitnami/apisix/deps /usr/local/apisix
-      ln -sf /opt/bitnami/apisix/ui /usr/local/apisix
-      ln -sf /opt/bitnami/apisix/openresty/luajit/share/lua/*/apisix /usr/local/apisix
-      mkdir -p /usr/local/apisix/logs
     {{- if .context.Values.usePasswordFiles }}
       {{- if .context.Values.controlPlane.enabled }}
       export APISIX_ADMIN_API_TOKEN="$(< $APISIX_ADMIN_API_TOKEN_FILE)"
