diff --git a/server/src/main/resources/org/opensearch/bootstrap/security.policy b/server/src/main/resources/org/opensearch/bootstrap/security.policy
--- a/server/src/main/resources/org/opensearch/bootstrap/security.policy
+++ b/server/src/main/resources/org/opensearch/bootstrap/security.policy
@@ -49,6 +49,8 @@ grant codeBase "${codebase.opensearch}" {
   permission java.lang.RuntimePermission "getClassLoader";
   // needed for SPI class loading
   permission java.lang.RuntimePermission "accessDeclaredMembers";
+  // needed for IsoCalendarDataProvider SPI registration (java.locale.providers=SPI,COMPAT)
+  permission java.lang.RuntimePermission "localeServiceProvider";
   permission org.opensearch.secure_sm.ThreadContextPermission "markAsSystemContext";
   permission org.opensearch.secure_sm.ThreadContextPermission "stashWithOrigin";
 };
