From 410a91df4b8cf6a5dc86a3fdd72975bd824f4436 Mon Sep 17 00:00:00 2001
From: Ben Tasker <2900301+bentasker@users.noreply.github.com>
Date: Fri, 13 Jun 2025 16:27:28 +0100
Subject: [PATCH] feat: upgrade kafka to remediate GHSA-vgq5-3255-v292

Signed-off-by: Ben Tasker <2900301+bentasker@users.noreply.github.com>
---
 build.gradle | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build.gradle b/build.gradle
index a0e095da..69aa7c69 100644
--- a/build.gradle
+++ b/build.gradle
@@ -25,7 +25,7 @@ buildscript {
         opensearch_build = version_tokens[0] + '.0'
 
         common_utils_version = System.getProperty("common_utils.version", '2.19.0.0')
-        kafka_version  = '3.7.1'
+        kafka_version  = '3.9.1'
         open_saml_version = '4.3.2'
         one_login_java_saml = '2.9.0'
         jjwt_version = '0.12.6'
-- 
2.43.0

