From 08348c8f7fc6e603b466de3577a1b1f756eda0de Mon Sep 17 00:00:00 2001
From: Debasish Biswas <debasishbsws.dev@gmail.com>
Date: Tue, 1 Jul 2025 13:14:13 +0530
Subject: force resolution for beanutils to fix CVE,
 update dependencies for Ghidra 12.0.1

diff --git a/Ghidra/Debug/Debugger-rmi-trace/Module.manifest b/Ghidra/Debug/Debugger-rmi-trace/Module.manifest
index 22c6608a..58f4193e 100644
--- a/Ghidra/Debug/Debugger-rmi-trace/Module.manifest
+++ b/Ghidra/Debug/Debugger-rmi-trace/Module.manifest
@@ -1,4 +1,4 @@
-MODULE FILE LICENSE: pypkg/dist/protobuf-6.31.0-py3-none-any.whl BSD-3-GOOGLE
+MODULE FILE LICENSE: pypkg/dist/protobuf-6.31.1-py3-none-any.whl BSD-3-GOOGLE
 MODULE FILE LICENSE: pypkg/dist/psutil-5.9.8.tar.gz BSD-3-PSUTIL
 MODULE FILE LICENSE: pypkg/dist/setuptools-80.9.0-py3-none-any.whl MIT
 MODULE FILE LICENSE: pypkg/dist/wheel-0.45.1-py3-none-any.whl MIT
diff --git a/Ghidra/Debug/Debugger-rmi-trace/build.gradle b/Ghidra/Debug/Debugger-rmi-trace/build.gradle
index 8fda8790..d0d56201 100644
--- a/Ghidra/Debug/Debugger-rmi-trace/build.gradle
+++ b/Ghidra/Debug/Debugger-rmi-trace/build.gradle
@@ -73,7 +73,7 @@ tasks.assemblePyPackage {
 	}
 }

-distributePyDep("protobuf-6.31.0-py3-none-any.whl")
+distributePyDep("protobuf-6.31.1-py3-none-any.whl")
 distributePyDep("psutil-5.9.8.tar.gz")
 distributePyDep("setuptools-80.9.0-py3-none-any.whl")
 distributePyDep("wheel-0.45.1-py3-none-any.whl")
diff --git a/Ghidra/Features/BSim/Module.manifest b/Ghidra/Features/BSim/Module.manifest
index 7109e133b4..69018f0fb5 100755
--- a/Ghidra/Features/BSim/Module.manifest
+++ b/Ghidra/Features/BSim/Module.manifest
@@ -1,6 +1,6 @@
 ##MODULE IP: Oxygen Icons - LGPL 3.0
 MODULE FILE LICENSE: postgresql-15.13.tar.gz Postgresql License
-MODULE FILE LICENSE: lib/postgresql-42.7.6.jar PostgresqlJDBC License
+MODULE FILE LICENSE: lib/postgresql-42.7.7.jar PostgresqlJDBC License
 MODULE FILE LICENSE: lib/commons-dbcp2-2.9.0.jar Apache License 2.0
 MODULE FILE LICENSE: lib/commons-pool2-2.11.1.jar Apache License 2.0
 MODULE FILE LICENSE: lib/commons-logging-1.2.jar Apache License 2.0
diff --git a/Ghidra/Features/BSim/build.gradle b/Ghidra/Features/BSim/build.gradle
index 0c88c9a329..f070d6c171 100755
--- a/Ghidra/Features/BSim/build.gradle
+++ b/Ghidra/Features/BSim/build.gradle
@@ -33,7 +33,7 @@ dependencies {
 	api project(":Decompiler")
 	api project(":CodeCompare")

-	api "org.postgresql:postgresql:42.7.6"
+	api "org.postgresql:postgresql:42.7.7"
     api "org.apache.commons:commons-dbcp2:2.9.0"
     api "org.apache.commons:commons-pool2:2.11.1"
     api "commons-logging:commons-logging:1.2"
diff --git a/Ghidra/Features/GhidraServer/Module.manifest b/Ghidra/Features/GhidraServer/Module.manifest
index 070fe00556..8f9897a3e2 100644
--- a/Ghidra/Features/GhidraServer/Module.manifest
+++ b/Ghidra/Features/GhidraServer/Module.manifest
@@ -1,2 +1,2 @@
-DATA SEARCH IGNORE DIR: yajsw-stable-13.12
+DATA SEARCH IGNORE DIR: yajsw-stable-13.13
 EXCLUDE FROM GHIDRA JAR: true
diff --git a/Ghidra/Features/GhidraServer/build.gradle b/Ghidra/Features/GhidraServer/build.gradle
index eef2f6ad4b..a1d5e44eec 100644
--- a/Ghidra/Features/GhidraServer/build.gradle
+++ b/Ghidra/Features/GhidraServer/build.gradle
@@ -21,7 +21,7 @@ apply plugin: 'eclipse'

 eclipse.project.name = 'Features GhidraServer'

-def yajswRelease = "yajsw-stable-13.12"
+def yajswRelease = "yajsw-stable-13.13"

 configurations {
 	runGhidraServer
diff --git a/build.gradle b/build.gradle
index 159eb7dd7b..6ad9ce66e9 100644
--- a/build.gradle
+++ b/build.gradle
@@ -117,6 +117,16 @@ clean {
 	delete "$buildDir"
 }

+// Pin beanutils versions to CVE free versions
+subprojects {
+    configurations.all {
+        resolutionStrategy {
+            force 'commons-beanutils:commons-beanutils:1.11.0'
+            force 'commons-beanutils:commons-beanutils-core:1.11.0'
+        }
+    }
+}
+
 /*********************************************************************************
  * Throws a GradleException if the current Gradle version is outside of the supported
  * Gradle version range defined in application.properties.
diff --git a/gradle/support/fetchDependencies.gradle b/gradle/support/fetchDependencies.gradle
index f3b0ef87b5..a6a6ad01ae 100644
--- a/gradle/support/fetchDependencies.gradle
+++ b/gradle/support/fetchDependencies.gradle
@@ -90,9 +90,9 @@ ext.deps = [
 		destination: FLAT_REPO_DIR
 	],
 	[
-		name: "yajsw-stable-13.12.zip",
-		url: "https://sourceforge.net/projects/yajsw/files/yajsw/yajsw-stable-13.12/yajsw-stable-13.12.zip",
-		sha256: "c6fc59815d3800d14ec977926a8afd3f606a0ebd74d2cfd60601677466edeaa2",
+		name: "yajsw-stable-13.13.zip",
+		url: "https://sourceforge.net/projects/yajsw/files/yajsw/yajsw-stable-13.13/yajsw-stable-13.13.zip",
+		sha256: "13fd79f2bdbf91dd9ac0c157c86440240eced2f21488a056e0843d2b8d459772",
 		destination: file("${DEPS_DIR}/GhidraServer")
 	],
 	[
@@ -212,9 +212,9 @@ ext.deps = [
 		}
 	],
 	[
-		name: "protobuf-6.31.0-py3-none-any.whl",
-		url: "https://files.pythonhosted.org/packages/ee/01/1ed1d482960a5718fd99c82f6d79120181947cfd4667ec3944d448ed44a3/protobuf-6.31.0-py3-none-any.whl",
-		sha256: "6ac2e82556e822c17a8d23aa1190bbc1d06efb9c261981da95c71c9da09e9e23",
+		name: "protobuf-6.31.1-py3-none-any.whl",
+		url: "https://files.pythonhosted.org/packages/f7/af/ab3c51ab7507a7325e98ffe691d9495ee3d3aa5f589afad65ec920d39821/protobuf-6.31.1-py3-none-any.whl",
+		sha256: "720a6c7e6b77288b85063569baae8536671b39f15cc22037ec7045658d80489e",
 		destination: file("${DEPS_DIR}/Debugger-rmi-trace/")
 	],
 	[
