From 08348c8f7fc6e603b466de3577a1b1f756eda0de Mon Sep 17 00:00:00 2001
From: Debasish Biswas <debasishbsws.dev@gmail.com>
Date: Tue, 1 Jul 2025 13:14:13 +0530
Subject: force resolution for beanutils to fix CVE,
 update dependencies for Ghidra 12.0.3

diff --git a/build.gradle b/build.gradle
index 159eb7dd7b..6ad9ce66e9 100644
--- a/build.gradle
+++ b/build.gradle
@@ -117,6 +117,16 @@ clean {
 	delete "$buildDir"
 }
 
+// Pin beanutils versions to CVE free versions
+subprojects {
+    configurations.all {
+        resolutionStrategy {
+            force 'commons-beanutils:commons-beanutils:1.11.0'
+            force 'commons-beanutils:commons-beanutils-core:1.11.0'
+        }
+    }
+}
+
 /*********************************************************************************
  * Throws a GradleException if the current Gradle version is outside of the supported
  * Gradle version range defined in application.properties.
