--- a/virt/firmware/efi/certs.py
+++ b/virt/firmware/efi/certs.py
@@ -13,19 +13,12 @@ from cryptography.hazmat.primitives import hashes, serialization
 from cryptography.hazmat.primitives.asymmetric import rsa


-if sys.version_info >= (3, 13):
-    import importlib.resources
-
-    def resource_filename(module, filename):
-        """
-        Mimics pkg_resources.resource_filename in a way.
-
-        Enters the context manager provided by importlib.resources.path
-        and never exits it.
-        """
-        return str(importlib.resources.path(module, filename).__enter__())
-else:
-    from pkg_resources import resource_filename
+import importlib.resources
+
+
+def resource_filename(module, filename):
+    """ Return filesystem path for a package resource. """
+    return str(importlib.resources.files(module).joinpath(filename))


 # redhat: PK + KEK key
--- a/virt/firmware/varstore/aws.py
+++ b/virt/firmware/varstore/aws.py
@@ -20,10 +20,7 @@ try:
     from crc32c import crc32c
 except ModuleNotFoundError:
     from virt.firmware.misc import crc32c
-if sys.version_info >= (3, 13):
-    import importlib.resources
-else:
-    from pkg_resources import resource_filename
+import importlib.resources


 zdict_v0_modname = 'virt.firmware'
@@ -34,15 +31,11 @@ zdict_v0_blob = None
 def zdict_v0():
     global zdict_v0_blob
     if zdict_v0_blob is None:
-        if sys.version_info >= (3, 13):
-            zdict_v0_blob = importlib.resources.read_binary(
-                zdict_v0_modname,
-                zdict_v0_filename,
-            )
-        else:
-            path = resource_filename(zdict_v0_modname, zdict_v0_filename)
-            with open(path, "rb") as f:
-                zdict_v0_blob = f.read()
+        zdict_v0_blob = (
+            importlib.resources.files(zdict_v0_modname)
+            .joinpath(zdict_v0_filename)
+            .read_bytes()
+        )
     return zdict_v0_blob
