diff --git a/CMakeLists.txt b/CMakeLists.txt
index 313b85d..7a43d48 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -159,13 +159,12 @@ set(PT_LIBS
     "libtorch_cuda.so"
     "libtorch_cuda_linalg.so"
     "libtorch_global_deps.so"
-    "libjpeg.so.62"
 )
 
 if (${TRITON_PYTORCH_ENABLE_TORCHVISION})
   set(PT_LIBS
       ${PT_LIBS}
-      $<IF:$<BOOL:${RHEL_BUILD}>,libtorchvision.so,libtorchvision.so.1>
+      "libtorchvision.so"
   )
 endif() # TRITON_PYTORCH_ENABLE_TORCHVISION
 
@@ -382,8 +381,8 @@ if (${TRITON_PYTORCH_DOCKER_BUILD})
 
   if (${TRITON_PYTORCH_ENABLE_TORCHVISION})
     set(TRITON_PYTORCH_LIBS
-    ${TRITON_PYTORCH_LIBS}
-    "${CMAKE_CURRENT_BINARY_DIR}/$<IF:$<BOOL:${RHEL_BUILD}>,libtorchvision.so,libtorchvision.so.1>")
+        ${TRITON_PYTORCH_LIBS}
+        "${CMAKE_CURRENT_BINARY_DIR}/libtorchvision.so")
   endif() # TRITON_PYTORCH_ENABLE_TORCHVISION
 
   if (${TRITON_PYTORCH_ENABLE_TORCHTRT})
@@ -514,18 +513,27 @@ else()
       ${PT_LIB_PATHS}
     DESTINATION ${CMAKE_INSTALL_PREFIX}/backends/pytorch
   )
-
-  FOREACH(plib ${PT_LIBS})
+  # Manually install OpenCV shared libraries when not using Docker-based build
+  FOREACH(opencvlib ${OPENCV_LIBS})
     install(
-      CODE
-        "EXECUTE_PROCESS(
-          COMMAND patchelf --set-rpath \$ORIGIN ${plib}
-          RESULT_VARIABLE PATCHELF_STATUS
-          WORKING_DIRECTORY ${CMAKE_INSTALL_PREFIX}/backends/pytorch)
-        if(PATCHELF_STATUS AND NOT PATCHELF_STATUS EQUAL 0)
-          message(FATAL_ERROR \"FAILED: to run patchelf\")
-        endif()"
+      FILES
+        "${TRITON_PYTORCH_LIB_PATHS}/${opencvlib}"
+      DESTINATION ${CMAKE_INSTALL_PREFIX}/backends/pytorch
     )
+  ENDFOREACH()
+
+  FOREACH(plib ${PT_LIBS})
+    # install(
+    #   CODE
+    #     "EXECUTE_PROCESS(
+    #       COMMAND patchelf --set-rpath \$ORIGIN ${plib}
+    #       RESULT_VARIABLE PATCHELF_STATUS
+    #       WORKING_DIRECTORY ${CMAKE_INSTALL_PREFIX}/backends/pytorch)
+    #     if(PATCHELF_STATUS AND NOT PATCHELF_STATUS EQUAL 0)
+    #       message(FATAL_ERROR \"FAILED: to run patchelf\")
+    #     endif()"
+    # )
+    message(WARNING "Skipping patchelf for ${plib} since TRITON_PYTORCH_DOCKER_BUILD is OFF")
   ENDFOREACH(plib)
 endif() # TRITON_PYTORCH_DOCKER_BUILD
 
@@ -540,6 +548,17 @@ install(
     ${INSTALL_CONFIGDIR}
 )
 
+if (${TRITON_PYTORCH_ENABLE_TORCHTRT})
+  if (EXISTS "/usr/bin/torchtrtc")
+    install(
+      FILES /usr/bin/torchtrtc
+      DESTINATION ${CMAKE_INSTALL_PREFIX}/backends/pytorch
+    )
+  else()
+    message(WARNING "TorchTRT is enabled but /usr/bin/torchtrtc was not found")
+  endif()
+endif()
+
 install(
   FILES
     src/model.py
