diff --git a/modules/repository-azure/build.gradle b/modules/repository-azure/build.gradle
index d093816acd45..7ed290504f8c 100644
--- a/modules/repository-azure/build.gradle
+++ b/modules/repository-azure/build.gradle
@@ -56,11 +56,11 @@ dependencies {
   api "com.fasterxml.woodstox:woodstox-core:${versions.woodstox}"
 
   // netty
-  api "io.netty:netty-codec-dns:${versions.netty}"
-  api "io.netty:netty-codec-http2:${versions.netty}"
-  api "io.netty:netty-codec-socks:${versions.netty}"
-  api "io.netty:netty-handler-proxy:${versions.netty}"
-  api "io.netty:netty-resolver-dns:${versions.netty}"
+  api "io.netty:netty-codec-dns:4.1.109.Final"
+  api "io.netty:netty-codec-http2:4.1.109.Final"
+  api "io.netty:netty-codec-socks:4.1.109.Final"
+  api "io.netty:netty-handler-proxy:4.1.109.Final"
+  api "io.netty:netty-resolver-dns:4.1.109.Final"
 
   // reactor
   api "io.projectreactor.netty:reactor-netty-core:${versions.reactorNetty}"
