diff --git a/modules/transport-netty4/build.gradle b/modules/transport-netty4/build.gradle
index 14b0b57da3b0..5891b4e60750 100644
--- a/modules/transport-netty4/build.gradle
+++ b/modules/transport-netty4/build.gradle
@@ -37,14 +37,14 @@ dependencies {
   api project(":libs:elasticsearch-ssl-config")
 
   // network stack
-  api "io.netty:netty-buffer:${versions.netty}"
-  api "io.netty:netty-codec:${versions.netty}"
-  api "io.netty:netty-codec-http:${versions.netty}"
-  api "io.netty:netty-common:${versions.netty}"
-  api "io.netty:netty-handler:${versions.netty}"
-  api "io.netty:netty-resolver:${versions.netty}"
-  api "io.netty:netty-transport:${versions.netty}"
-  api "io.netty:netty-transport-native-unix-common:${versions.netty}"
+  api "io.netty:netty-buffer:4.1.109.Final"
+  api "io.netty:netty-codec:4.1.109.Final"
+  api "io.netty:netty-codec-http:4.1.109.Final"
+  api "io.netty:netty-common:4.1.109.Final"
+  api "io.netty:netty-handler:4.1.109.Final"
+  api "io.netty:netty-resolver:4.1.109.Final"
+  api "io.netty:netty-transport:4.1.109.Final"
+  api "io.netty:netty-transport-native-unix-common:4.1.109.Final"
 
   testImplementation project(':modules:rest-root')
 }
