diff --git a/x-pack/plugin/core/build.gradle b/x-pack/plugin/core/build.gradle
index 0c65c7e4b6d2..ccafb450cafd 100644
--- a/x-pack/plugin/core/build.gradle
+++ b/x-pack/plugin/core/build.gradle
@@ -51,7 +51,7 @@ dependencies {
 
   // security deps
   api 'com.unboundid:unboundid-ldapsdk:6.0.3'
-  api "com.nimbusds:nimbus-jose-jwt:9.23"
+  api "com.nimbusds:nimbus-jose-jwt:9.37.2"
 
   implementation project(":x-pack:plugin:core:template-resources")
 
