diff --git a/gradle/dependencies.gradle b/gradle/dependencies.gradle
index faef12d233..a941bd4b01 100644
--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -148,7 +148,7 @@ versions += [
   lz4: "1.8.0",
   mavenArtifact: "3.9.6",
   metrics: "2.2.0",
-  netty: "4.1.110.Final",
+  netty: "4.1.125.Final",
   opentelemetryProto: "1.0.0-alpha",
   protobuf: "3.25.5", // a dependency of opentelemetryProto
   pcollections: "4.0.1",
