From d480ccff1cb5eadad16fbacf54f11d15cdfeaa1e Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Wed, 20 Aug 2025 00:23:14 +0100
Subject: [PATCH] Add quiche.patch

---
 bazel/repositories.bzl | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/bazel/repositories.bzl b/bazel/repositories.bzl
index 211e9d8060..9900c53964 100644
--- a/bazel/repositories.bzl
+++ b/bazel/repositories.bzl
@@ -775,6 +775,8 @@ def _com_github_google_quiche():
         name = "com_github_google_quiche",
         patch_cmds = ["find quiche/ -type f -name \"*.bazel\" -delete"],
         build_file = "@envoy//bazel/external:quiche.BUILD",
+        patches = ["@envoy//bazel/external:quiche.patch"],
+        patch_args = ["-p1"],
     )
 
 def _com_googlesource_googleurl():
-- 
2.48.1

