From 902c56bcbe7b90b0aaa53b9148d117353fe2ec39 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Tue, 3 Mar 2026 22:01:29 +0000
Subject: [PATCH] Also set extra-jlink-flags on the legacy jre image

---
 make/Images.gmk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/make/Images.gmk b/make/Images.gmk
index 22e3e43cb1f..9243b15faa0 100644
--- a/make/Images.gmk
+++ b/make/Images.gmk
@@ -99,6 +99,7 @@ endif
 
 ifneq ($(JLINK_USER_EXTRA_FLAGS), )
   JLINK_JDK_EXTRA_OPTS += $(JLINK_USER_EXTRA_FLAGS)
+  JLINK_JRE_EXTRA_OPTS += $(JLINK_USER_EXTRA_FLAGS)
 endif
 
 $(eval $(call SetupExecute, jlink_jdk, \
-- 
2.51.0

