diff --git a/cmake/warnings.cmake b/cmake/warnings.cmake
index 3a0a5ebb..ce36025c 100644
--- a/cmake/warnings.cmake
+++ b/cmake/warnings.cmake
@@ -47,3 +47,6 @@ no_warning(thread-safety-negative) # experimental flag, too many false positives
 no_warning(unsafe-buffer-usage) # too aggressive
 no_warning(switch-default) # conflicts with "defaults in a switch covering all enum values"
 # TODO Enable conversion, sign-conversion, double-promotion warnings.
+no_warning(unused-result)
+no_warning(deprecated-declarations)
+no_warning(reorder-ctor)
