--- a/pkg/version/version.go	2024-11-27 15:07:23
+++ b/pkg/version/version.go	2024-11-27 14:58:47
@@ -42,7 +42,9 @@
 func FromString(versionString string) CiliumVersion {
 	// string to parse: "0.13.90 a722bdb 2018-01-09T22:32:37+01:00 go version go1.9 linux/amd64"
 	fields := strings.Split(versionString, " ")
-	if len(fields) != 7 {
+	// Our go-fips version string can contain spaces (e.g. `go-X.XX.X X:systemcrypto`)
+	// in which case the length of the fields will be 8.
+	if len(fields) < 7 || len(fields) > 8 {
 		return CiliumVersion{}
 	}
 
@@ -50,8 +52,8 @@
 		Version:          fields[0],
 		Revision:         fields[1],
 		AuthorDate:       fields[2],
-		GoRuntimeVersion: fields[5],
-		Arch:             fields[6],
+		GoRuntimeVersion: strings.Join(fields[5:len(fields)-1], " "),
+		Arch:             fields[len(fields)-1],
 	}
 	return cver
 }
