From f18b9f63ddd2dd1b293d99e03bcad54b255836f6 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Wed, 31 Dec 2025 12:23:23 +0000
Subject: [PATCH] libbson: add missing defines

---
 src/third_party/libbson/BUILD.bazel | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/third_party/libbson/BUILD.bazel b/src/third_party/libbson/BUILD.bazel
index 145704ccb8b..15e03dbe1d7 100644
--- a/src/third_party/libbson/BUILD.bazel
+++ b/src/third_party/libbson/BUILD.bazel
@@ -91,5 +91,7 @@ mongo_cc_library(
     }),
     local_defines = [
         "BSON_COMPILATION",
+        "_XOPEN_SOURCE=700",
+        "_GNU_SOURCE",
     ],
 )
-- 
2.51.0

