From 58d6cf617e6ce8c0c91bd15c8fdb2965b527b5e9 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Wed, 31 Dec 2025 13:08:06 +0000
Subject: [PATCH] util: underlinked dns_query

---
 src/mongo/util/BUILD.bazel | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/mongo/util/BUILD.bazel b/src/mongo/util/BUILD.bazel
index cbae5b3d932..e5798448636 100644
--- a/src/mongo/util/BUILD.bazel
+++ b/src/mongo/util/BUILD.bazel
@@ -318,6 +318,9 @@ mongo_cc_library(
 
 mongo_cc_library(
     name = "dns_query",
+    linkopts = [
+        "-lresolv",
+    ],
     srcs = [
         "//src/mongo/util:dns_query.cpp",
     ],
-- 
2.51.0

