diff --git a/spark-operator/build.gradle b/spark-operator/build.gradle
index c63b8de..fips0001 100644
--- a/spark-operator/build.gradle
+++ b/spark-operator/build.gradle
@@ -67,6 +67,7 @@ shadowJar {
   zip64 = true
   mergeServiceFiles()
   transform(com.github.jengelman.gradle.plugins.shadow.transformers.Log4j2PluginsCacheFileTransformer)
+  exclude("org/bouncycastle/**")
   exclude("org/apache/spark/ui/static/**")
 }

