From d2f7bf9fae4b666373c6c60748841d6e7aefd24c Mon Sep 17 00:00:00 2001
From: Krishnan Prashanth <kprashanth@nvidia.com>
Date: Thu, 17 Oct 2024 17:39:40 -0700
Subject: [PATCH] updating pybind11 version

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ee209b5b..fc078f2c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -88,8 +88,8 @@ FetchContent_MakeAvailable(repo-common repo-core repo-backend)
 FetchContent_Declare(
   pybind11
   GIT_REPOSITORY "https://github.com/pybind/pybind11"
-  # COMMIT ID for v2.10.0
-  GIT_TAG "aa304c9c7d725ffb9d10af08a3b34cb372307020"
+  # COMMIT ID for v2.12.0
+  GIT_TAG "3e9dfa2866941655c56877882565e7577de6fc7b"
   GIT_SHALLOW ON
 )
 FetchContent_MakeAvailable(pybind11)
