From 9b099562b7463649c96d4f03a51b96e916781ef9 Mon Sep 17 00:00:00 2001
From: Ajay K <ajaykemparaj@gmail.com>
Date: Thu, 6 Mar 2025 04:31:06 +0000
Subject: [PATCH 5/7] set gofips as latest always

---
 src/cmd/dist/build.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cmd/dist/build.go b/src/cmd/dist/build.go
index cadd5c9aec..0e6ca93986 100644
--- a/src/cmd/dist/build.go
+++ b/src/cmd/dist/build.go
@@ -188,7 +188,7 @@ func xinit() {
 
 	b = os.Getenv("GOFIPS140")
 	if b == "" {
-		b = "off"
+		b = "latest"
 	}
 	gofips140 = b
 
-- 
2.48.1

