From 66c834b25c967273eca9fc453b5335f31badac13 Mon Sep 17 00:00:00 2001
From: Ajay K <ajaykemparaj@gmail.com>
Date: Wed, 5 Mar 2025 23:35:57 +0000
Subject: [PATCH] default go fips enabled

---
 src/internal/buildcfg/cfg.go | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/src/internal/buildcfg/cfg.go b/src/internal/buildcfg/cfg.go
index 5ae4c0c7ad..aafeeac679 100644
--- a/src/internal/buildcfg/cfg.go
+++ b/src/internal/buildcfg/cfg.go
@@ -72,16 +72,7 @@ func goamd64() int {
 }
 
 func gofips140() string {
-	v := envOr("GOFIPS140", DefaultGOFIPS140)
-	switch v {
-	case "off", "latest", "inprocess", "certified":
-		return v
-	}
-	if isFIPSVersion(v) {
-		return v
-	}
-	Error = fmt.Errorf("invalid GOFIPS140: must be off, latest, inprocess, certified, or vX.Y.Z")
-	return DefaultGOFIPS140
+	return "latest"
 }
 
 // isFIPSVersion reports whether v is a valid FIPS version,
-- 
2.48.1

