From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: James Page <james.page@chainguard.dev>
Date: Tue, 29 Apr 2025 11:27:30 +0100
Subject: [PATCH] ed25519: check rand beforing using boring.

---
 src/crypto/ed25519/ed25519.go | 25 +++++++++++--------------
 1 file changed, 11 insertions(+), 14 deletions(-)

diff --git a/src/crypto/ed25519/ed25519.go b/src/crypto/ed25519/ed25519.go
index 6476bfbe896d6c..a3152f77e0dbc6 100644
--- a/src/crypto/ed25519/ed25519.go
+++ b/src/crypto/ed25519/ed25519.go
@@ -140,21 +140,18 @@ func GenerateKey(rand io.Reader) (PublicKey, PrivateKey, error) {
 	if rand == nil {
 		rand = cryptorand.Reader
 	}
-	if boring.Enabled && boring.SupportsEd25519() {
-		if rand == boring.RandReader {
-			priv, err := boring.GenerateKeyEd25519()
-			if err != nil {
-				return nil, nil, err
-			}
-			privData, err := priv.Bytes()
-			if err != nil {
-				return nil, nil, err
-			}
-			privKey := PrivateKey(privData)
-			pubKey := privKey.Public().(PublicKey)
-			return pubKey, privKey, err
+	if boring.Enabled && rand == boring.RandReader && boring.SupportsEd25519() {
+		priv, err := boring.GenerateKeyEd25519()
+		if err != nil {
+			return nil, nil, err
+		}
+		privData, err := priv.Bytes()
+		if err != nil {
+			return nil, nil, err
 		}
-		boring.UnreachableExceptTests()
+		privKey := PrivateKey(privData)
+		pubKey := privKey.Public().(PublicKey)
+		return pubKey, privKey, err
 	}
 
 	seed := make([]byte, SeedSize)
