diff --git a/torch/csrc/jit/operator_upgraders/utils.cpp b/torch/csrc/jit/operator_upgraders/utils.cpp
index 2cfd7c0..57fa245 100644
--- a/torch/csrc/jit/operator_upgraders/utils.cpp
+++ b/torch/csrc/jit/operator_upgraders/utils.cpp
@@ -7,6 +7,7 @@
 #include <regex>
 #include <string>
 #include <vector>
+#include <cstdint>

 namespace torch::jit {

diff --git a/torch/csrc/jit/operator_upgraders/utils.h b/torch/csrc/jit/operator_upgraders/utils.h
index 8228b69..4ffc6e9 100644
--- a/torch/csrc/jit/operator_upgraders/utils.h
+++ b/torch/csrc/jit/operator_upgraders/utils.h
@@ -4,6 +4,7 @@
 #include <torch/csrc/jit/operator_upgraders/version_map.h>
 #include <string>
 #include <vector>
+#include <cstdint>

 namespace torch::jit {
    