From 831a0ce5fd7f29ca47cf030383b13203b9a1e85a Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Fri, 23 Jan 2026 22:45:34 +0000
Subject: [PATCH] fips: always set ssl_compliance_policy_fips_202205

---
 source/common/tls/context_impl.cc | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/source/common/tls/context_impl.cc b/source/common/tls/context_impl.cc
index ff17b669f9..1fbea2c987 100644
--- a/source/common/tls/context_impl.cc
+++ b/source/common/tls/context_impl.cc
@@ -187,8 +187,6 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const Envoy::Ssl::ContextConfig& c
     }
   }
 
-  const bool fips_mode = FIPS_mode();
-
 #ifdef BORINGSSL_FIPS
   if (!capabilities_.is_fips_compliant) {
     creation_status = absl::InvalidArgumentError(
@@ -355,14 +353,6 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const Envoy::Ssl::ContextConfig& c
     }
   }
 
-  // Compliance policy must be applied last to have a defined behavior.
-  if (const auto policy = config.compliancePolicy(); policy.has_value()) {
-    switch (policy.value()) {
-      using ProtoPolicy = envoy::extensions::transport_sockets::tls::v3::TlsParameters;
-    case ProtoPolicy::FIPS_202205:
-      if (!fips_mode) {
-        ENVOY_LOG(warn, "FIPS conformance policy applied on a non-FIPS build");
-      }
       for (auto& tls_context : tls_contexts_) {
         int rc = SSL_CTX_set_compliance_policy(tls_context.ssl_ctx_.get(),
                                                ssl_compliance_policy_fips_202205);
@@ -373,12 +363,6 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const Envoy::Ssl::ContextConfig& c
           return;
         }
       }
-      break;
-    default:
-      creation_status = absl::InvalidArgumentError("Unknown compliance policy");
-      return;
-    }
-  }
 }
 
 void ContextImpl::keylogCallback(const SSL* ssl, const char* line) {
-- 
2.51.0

