diff --git a/utils/build-presets.ini b/utils/build-presets.ini
index 27f68b58..3a68ad3e 100644
--- a/utils/build-presets.ini
+++ b/utils/build-presets.ini
@@ -870,7 +870,8 @@ build-swift-stdlib-unittest-extra
 # Executes the lit tests for the installable package that is created
 # Assumes the swift-integration-tests repo is checked out

-test-installable-package
+# We do not release the package, so no point testing it
+#test-installable-package

 # Build the benchmarks against the toolchain.
 toolchain-benchmarks
@@ -3040,3 +3041,18 @@ skip-test-cmark
 skip-test-swift
 skip-build-benchmarks
 skip-test-foundation
+
+[preset: wolfi]
+mixin-preset=buildbot_linux,no_assertions,no_test
+build-ninja=0
+extra-cmake-options=
+       -DLLVM_USE_LINKER=lld
+       -DBUILD_TESTING:BOOL=NO
+       -DSWIFT_INCLUDE_TESTS:BOOL=NO
+       -DSWIFT_INCLUDE_TEST_BINARIES:BOOL=NO
+       -DCOMPILER_RT_BUILD_ORC:BOOL=NO
+       -DPython3_FIND_UNVERSIONED_NAMES=FIRST
+libicu=0
+#llvm-targets-to-build=host
+skip-build-curl
+skip-build-zlib
