diff --git a/api/build.gradle b/api/build.gradle
--- a/api/build.gradle
+++ b/api/build.gradle
@@ -79,6 +79,9 @@
     implementation libs.logback.core
     implementation libs.logback.classic

+    implementation libs.jackson.core
+    implementation libs.jackson.databind
+
     implementation libs.modelcontextprotocol.spring.webflux
     implementation libs.victools.jsonschema.generator

diff --git a/gradle/libs.versions.toml b/gradle/libs.versions.toml
--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -3,6 +3,7 @@
 nimbus-jose-jwt = '10.0.2'
 netty = '4.1.130.Final'
 logback = '1.5.25'
+jackson = '2.21.1'

 aws-msk-auth = '2.3.0'
 azure-identity = '1.15.4'
@@ -141,3 +142,5 @@
 netty-codec-http = {module = 'io.netty:netty-codec-http', version.ref = 'netty'}
 logback-core = {module = 'ch.qos.logback:logback-core', version.ref = 'logback'}
 logback-classic = {module = 'ch.qos.logback:logback-classic', version.ref = 'logback'}
+jackson-core = {module = 'com.fasterxml.jackson.core:jackson-core', version.ref = 'jackson'}
+jackson-databind = {module = 'com.fasterxml.jackson.core:jackson-databind', version.ref = 'jackson'}
