diff --git a/authentik/admin/api/system.py b/authentik/admin/api/system.py
index 027db43e7..f489f4174 100644
--- a/authentik/admin/api/system.py
+++ b/authentik/admin/api/system.py
@@ -27,9 +27,9 @@ from authentik.rbac.permissions import HasPermission
 
 def fips_enabled():
     try:
-        from authentik.enterprise.license import LicenseKey
+        # from authentik.enterprise.license import LicenseKey
 
-        return backend._fips_enabled if LicenseKey.get_total().status().is_valid else None
+        return backend._fips_enabled # if LicenseKey.get_total().status().is_valid else None
     except ModuleNotFoundError:
         return None
 
diff --git a/authentik/blueprints/tests/test_managed_app_config.py b/authentik/blueprints/tests/test_managed_app_config.py
index 784811035..92a9491bf 100644
--- a/authentik/blueprints/tests/test_managed_app_config.py
+++ b/authentik/blueprints/tests/test_managed_app_config.py
@@ -1,14 +1,14 @@
 from django.test import TestCase
 
 from authentik.blueprints.apps import ManagedAppConfig
-from authentik.enterprise.apps import EnterpriseConfig
+# from authentik.enterprise.apps import EnterpriseConfig
 from authentik.lib.utils.reflection import get_apps
 
 
 class TestManagedAppConfig(TestCase):
     def test_apps_use_managed_app_config(self):
         for app in get_apps():
-            if app.name.startswith("authentik.enterprise"):
-                self.assertIn(EnterpriseConfig, app.__class__.__bases__)
-            else:
-                self.assertIn(ManagedAppConfig, app.__class__.__bases__)
+            # if app.name.startswith("authentik.enterprise"):
+            #     self.assertIn(EnterpriseConfig, app.__class__.__bases__)
+            # else:
+            self.assertIn(ManagedAppConfig, app.__class__.__bases__)
diff --git a/authentik/core/api/groups.py b/authentik/core/api/groups.py
index 8675ccdf4..d76938501 100644
--- a/authentik/core/api/groups.py
+++ b/authentik/core/api/groups.py
@@ -267,14 +267,14 @@ class GroupViewSet(UsedByMixin, ModelViewSet):
     def get_ql_fields(self):
         from djangoql.schema import BoolField, StrField
 
-        from authentik.enterprise.search.fields import (
-            JSONSearchField,
-        )
+        # from authentik.enterprise.search.fields import (
+        #     JSONSearchField,
+        # )
 
         return [
             StrField(Group, "name"),
             BoolField(Group, "is_superuser", nullable=True),
-            JSONSearchField(Group, "attributes"),
+            # JSONSearchField(Group, "attributes"),
         ]
 
     def get_queryset(self):
diff --git a/authentik/core/api/users.py b/authentik/core/api/users.py
index 804101743..5a634ac0b 100644
--- a/authentik/core/api/users.py
+++ b/authentik/core/api/users.py
@@ -526,10 +526,10 @@ class UserViewSet(
     def get_ql_fields(self):
         from djangoql.schema import BoolField, StrField
 
-        from authentik.enterprise.search.fields import (
-            ChoiceSearchField,
-            JSONSearchField,
-        )
+        # from authentik.enterprise.search.fields import (
+        #     ChoiceSearchField,
+        #     JSONSearchField,
+        # )
 
         return [
             StrField(User, "username"),
@@ -537,8 +537,8 @@ class UserViewSet(
             StrField(User, "email"),
             StrField(User, "path"),
             BoolField(User, "is_active", nullable=True),
-            ChoiceSearchField(User, "type"),
-            JSONSearchField(User, "attributes"),
+            # ChoiceSearchField(User, "type"),
+            # JSONSearchField(User, "attributes"),
         ]
 
     def get_queryset(self):
diff --git a/authentik/events/api/events.py b/authentik/events/api/events.py
index 104d3a119..5ea0fea80 100644
--- a/authentik/events/api/events.py
+++ b/authentik/events/api/events.py
@@ -139,49 +139,49 @@ class EventViewSet(
     def get_ql_fields(self):
         from djangoql.schema import DateTimeField, IntField, StrField
 
-        from authentik.enterprise.search.fields import ChoiceSearchField, JSONSearchField
+        # from authentik.enterprise.search.fields import ChoiceSearchField, JSONSearchField
 
         return [
-            ChoiceSearchField(Event, "action"),
+            # ChoiceSearchField(Event, "action"),
             StrField(Event, "event_uuid"),
             StrField(Event, "app", suggest_options=True),
             StrField(Event, "client_ip"),
-            JSONSearchField(
-                Event,
-                "user",
-                fixed_structure=OrderedDict(
-                    pk=IntField(),
-                    username=StrField(),
-                    email=StrField(),
-                ),
-            ),
-            JSONSearchField(
-                Event,
-                "brand",
-                fixed_structure=OrderedDict(
-                    pk=StrField(),
-                    app=StrField(),
-                    name=StrField(),
-                    model_name=StrField(),
-                ),
-            ),
-            JSONSearchField(
-                Event,
-                "context",
-                fixed_structure=OrderedDict(
-                    http_request=JSONSearchField(
-                        Event,
-                        "context_http_request",
-                        fixed_structure=OrderedDict(
-                            args=JSONSearchField(Event, "context_http_request_args"),
-                            path=StrField(),
-                            method=StrField(),
-                            request_id=StrField(),
-                            user_agent=StrField(),
-                        ),
-                    ),
-                ),
-            ),
+            # JSONSearchField(
+            #     Event,
+            #     "user",
+            #     fixed_structure=OrderedDict(
+            #         pk=IntField(),
+            #         username=StrField(),
+            #         email=StrField(),
+            #     ),
+            # ),
+            # JSONSearchField(
+            #     Event,
+            #     "brand",
+            #     fixed_structure=OrderedDict(
+            #         pk=StrField(),
+            #         app=StrField(),
+            #         name=StrField(),
+            #         model_name=StrField(),
+            #     ),
+            # ),
+            # JSONSearchField(
+            #     Event,
+            #     "context",
+            #     fixed_structure=OrderedDict(
+            #         http_request=JSONSearchField(
+            #             Event,
+            #             "context_http_request",
+            #             fixed_structure=OrderedDict(
+            #                 args=JSONSearchField(Event, "context_http_request_args"),
+            #                 path=StrField(),
+            #                 method=StrField(),
+            #                 request_id=StrField(),
+            #                 user_agent=StrField(),
+            #             ),
+            #         ),
+            #     ),
+            # ),
             DateTimeField(Event, "created", suggest_options=True),
         ]
 
diff --git a/authentik/events/middleware.py b/authentik/events/middleware.py
index 7d13c0e6d..2e493502a 100644
--- a/authentik/events/middleware.py
+++ b/authentik/events/middleware.py
@@ -252,5 +252,5 @@ class AuditMiddleware:
             request,
             user=user,
             model=model_to_dict(instance),
-            **thread_kwargs,
+            **(thread_kwargs or {}),
         ).run()
diff --git a/authentik/providers/scim/models.py b/authentik/providers/scim/models.py
index 0339b3d77..211d23af0 100644
--- a/authentik/providers/scim/models.py
+++ b/authentik/providers/scim/models.py
@@ -141,13 +141,13 @@ class SCIMProvider(OutgoingSyncProvider, BackchannelProvider):
     )
 
     def scim_auth(self) -> AuthBase:
-        if self.auth_mode == SCIMAuthenticationMode.OAUTH:
-            try:
-                from authentik.enterprise.providers.scim.auth_oauth2 import SCIMOAuthAuth
+        # if self.auth_mode == SCIMAuthenticationMode.OAUTH:
+        #     try:
+        #         from authentik.enterprise.providers.scim.auth_oauth2 import SCIMOAuthAuth
 
-                return SCIMOAuthAuth(self)
-            except ImportError:
-                LOGGER.warning("Failed to import SCIM OAuth Client")
+        #         return SCIMOAuthAuth(self)
+        #     except ImportError:
+        #         LOGGER.warning("Failed to import SCIM OAuth Client")
         return SCIMTokenAuth(self)
 
     @property
diff --git a/authentik/root/settings.py b/authentik/root/settings.py
index 5345fb97f..f65e0e904 100644
--- a/authentik/root/settings.py
+++ b/authentik/root/settings.py
@@ -547,7 +547,7 @@ def _update_settings(app_path: str) -> None:
 
 # Attempt to load enterprise app, if available
 try:
-    importlib.import_module("authentik.enterprise.apps")
+    importlib.import_module("NO.ENTERPRISE.authentik.enterprise.apps")
     CONFIG.log("info", "Enabled authentik enterprise")
     TENANT_APPS.insert(TENANT_APPS.index("authentik.events"), "authentik.enterprise")
 except ImportError:
diff --git a/web/src/admin/admin-overview/charts/SyncStatusChart.ts b/web/src/admin/admin-overview/charts/SyncStatusChart.ts
index d74d56773..61ca7e3ac 100644
--- a/web/src/admin/admin-overview/charts/SyncStatusChart.ts
+++ b/web/src/admin/admin-overview/charts/SyncStatusChart.ts
@@ -122,32 +122,32 @@ export class SyncStatusChart extends AKChart<SummarizedSyncStatus[]> {
                 },
                 msg("SCIM Provider"),
             ),
-            await this.fetchStatus(
-                () => {
-                    return new ProvidersApi(DEFAULT_CONFIG).providersGoogleWorkspaceList();
-                },
-                (element) => {
-                    return new ProvidersApi(
-                        DEFAULT_CONFIG,
-                    ).providersGoogleWorkspaceSyncStatusRetrieve({
-                        id: element.pk,
-                    });
-                },
-                msg("Google Workspace Provider"),
-            ),
-            await this.fetchStatus(
-                () => {
-                    return new ProvidersApi(DEFAULT_CONFIG).providersMicrosoftEntraList();
-                },
-                (element) => {
-                    return new ProvidersApi(
-                        DEFAULT_CONFIG,
-                    ).providersMicrosoftEntraSyncStatusRetrieve({
-                        id: element.pk,
-                    });
-                },
-                msg("Microsoft Entra Provider"),
-            ),
+            // await this.fetchStatus(
+            //     () => {
+            //         return new ProvidersApi(DEFAULT_CONFIG).providersGoogleWorkspaceList();
+            //     },
+            //     (element) => {
+            //         return new ProvidersApi(
+            //             DEFAULT_CONFIG,
+            //         ).providersGoogleWorkspaceSyncStatusRetrieve({
+            //             id: element.pk,
+            //         });
+            //     },
+            //     msg("Google Workspace Provider"),
+            // ),
+            // await this.fetchStatus(
+            //     () => {
+            //         return new ProvidersApi(DEFAULT_CONFIG).providersMicrosoftEntraList();
+            //     },
+            //     (element) => {
+            //         return new ProvidersApi(
+            //             DEFAULT_CONFIG,
+            //         ).providersMicrosoftEntraSyncStatusRetrieve({
+            //             id: element.pk,
+            //         });
+            //     },
+            //     msg("Microsoft Entra Provider"),
+            // ),
             await this.fetchStatus(
                 () => {
                     return new SourcesApi(DEFAULT_CONFIG).sourcesLdapList();
