From 13661fbec970ff03b09d98408daa9a8c7351015c Mon Sep 17 00:00:00 2001
From: Ankush Pathak <ankush.pathak@chainguard.dev>
Date: Thu, 18 Dec 2025 16:48:32 +0000
Subject: [PATCH] Remediate GHSA-84h7-rjj3-6jx4 by bumping netty to
 4.1.129.Final

Signed-off-by: Ankush Pathak <ankush.pathak@chainguard.dev>
---
 modules/repository-azure/build.gradle | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/repository-azure/build.gradle b/modules/repository-azure/build.gradle
index c82f2a967b7..be995a29341 100644
--- a/modules/repository-azure/build.gradle
+++ b/modules/repository-azure/build.gradle
@@ -21,6 +21,7 @@ esplugin {
 
 versions << [
   'azureReactorNetty': '1.2.8',
+  'netty': '4.1.129.Final',
 ]
 
 dependencies {
-- 
2.52.0

