diff --git a/Cargo.toml b/Cargo.toml
index c179fb8..2251bf4 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -40,7 +40,7 @@ boring-sys = { version = "4", optional = true }
 ring = { version = "0.17", optional = true }
 
 # Enabled with 'tls-openssl'
-rustls-openssl = { version = "0.3", optional = true }
+rustls-openssl = { version = "0.3", features = ["fips"], default-features = false, optional = true }
 openssl = { version = "0.10", optional = true }
 
 anyhow = "1.0"
