diff --git a/src/dev/build/tasks/build_packages_task.ts b/src/dev/build/tasks/build_packages_task.ts
index f7d5206a9..fa95d23ad 100644
--- a/src/dev/build/tasks/build_packages_task.ts
+++ b/src/dev/build/tasks/build_packages_task.ts
@@ -334,6 +334,7 @@ export async function buildWebpackBundles({
   noCache?: boolean;
 }) {
   const options = [
+    ['--allow-root'],
     quiet ? ['--quiet'] : [],
     dist ? ['--dist'] : [],
     noCache ? ['--no-cache'] : [],
