diff --git a/bitnami/fluentd/templates/_helpers.tpl b/bitnami/fluentd/templates/_helpers.tpl
index 4ba6793643..de82603e0f 100644
--- a/bitnami/fluentd/templates/_helpers.tpl
+++ b/bitnami/fluentd/templates/_helpers.tpl
@@ -41,11 +41,6 @@ Create the name of the aggregator service account to use
 {{- end -}}
 {{- end -}}
 
-{{/* Check if there are rolling tags in the images */}}
-{{- define "fluentd.checkRollingTags" -}}
-{{- include "common.warnings.rollingTag" .Values.image }}
-{{- end -}}
-
 {{/*
 Validate data
 */}}
