diff --git a/bitnami/fluentd/templates/NOTES.txt b/bitnami/fluentd/templates/NOTES.txt
index 8646f759b8..9cfe87b5ae 100644
--- a/bitnami/fluentd/templates/NOTES.txt
+++ b/bitnami/fluentd/templates/NOTES.txt
@@ -2,8 +2,6 @@ CHART NAME: {{ .Chart.Name }}
 CHART VERSION: {{ .Chart.Version }}
 APP VERSION: {{ .Chart.AppVersion }}
 
-Did you know there are enterprise versions of the Bitnami catalog? For enhanced secure software supply chain features, unlimited pulls from Docker, LTS support, or application customization, see Bitnami Premium or Tanzu Application Catalog. See https://www.arrow.com/globalecs/na/vendors/bitnami for more information.
-
 ** Please be patient while the chart is being deployed **
 
 {{- if .Values.diagnosticMode.enabled }}
@@ -54,7 +52,4 @@ In order to replicate the container startup scripts execute this command:
 {{- end }}
 
 {{- include "fluentd.validateValues" . }}
-{{- include "fluentd.checkRollingTags" . -}}
 {{- include "common.warnings.resources" (dict "sections" (list "aggregator" "forwarder") "context" $) }}
-{{- include "common.warnings.modifiedImages" (dict "images" (list .Values.image) "context" $) }}
-{{- include "common.errors.insecureImages" (dict "images" (list .Values.image) "context" $) }}
diff --git a/bitnami/fluentd/templates/_helpers.tpl b/bitnami/fluentd/templates/_helpers.tpl
index 6cd3a97f49..de82603e0f 100644
--- a/bitnami/fluentd/templates/_helpers.tpl
+++ b/bitnami/fluentd/templates/_helpers.tpl
@@ -41,14 +41,6 @@ Create the name of the aggregator service account to use
 {{- end -}}
 {{- end -}}
 
-{{/* Check if there are rolling tags in the images */}}
-{{- define "fluentd.checkRollingTags" -}}
-{{- if and (contains "bitnami/" .Values.image.repository) (not (.Values.image.tag | toString | regexFind "-r\\d+$|sha256:")) }}
-WARNING: Rolling tag detected ({{ .Values.image.repository }}:{{ .Values.image.tag }}), please note that it is strongly recommended to avoid using rolling tags in a production environment.
-+info https://techdocs.broadcom.com/us/en/vmware-tanzu/application-catalog/tanzu-application-catalog/services/tac-doc/apps-tutorials-understand-rolling-tags-containers-index.html
-{{- end }}
-{{- end -}}
-
 {{/*
 Validate data
 */}}
diff --git a/bitnami/fluentd/values.yaml b/bitnami/fluentd/values.yaml
index 1782b15014..f7700cf11e 100644
--- a/bitnami/fluentd/values.yaml
+++ b/bitnami/fluentd/values.yaml
@@ -86,6 +86,7 @@ image:
   registry: docker.io
   repository: bitnami/fluentd
   tag: 1.18.0-debian-12-r21
+  digest: ""
   ## Specify a imagePullPolicy
   ## ref: https://kubernetes.io/docs/concepts/containers/images/#pre-pulled-images
   ##
