diff --git a/x-pack/plugin/ml/build.gradle b/x-pack/plugin/ml/build.gradle
index c13d178c3b2..6c7303ffe3d 100644
--- a/x-pack/plugin/ml/build.gradle
+++ b/x-pack/plugin/ml/build.gradle
@@ -90,7 +93,7 @@ dependencies {
   api project(':libs:elasticsearch-grok')
   api "net.sf.supercsv:super-csv:${versions.supercsv}"
   api "org.apache.commons:commons-math3:3.6.1"
-  nativeBundle("org.elasticsearch.ml:ml-cpp:${project.version}@zip") {
+  nativeBundle("org.elasticsearch.ml:ml-cpp:7.17.25@zip") {
     changing = true
   }
   testImplementation 'org.ini4j:ini4j:0.5.2'
