diff --git a/x-pack/plugin/identity-provider/build.gradle b/x-pack/plugin/identity-provider/build.gradle
index b15f28662324..5d4f677d16ee 100644
--- a/x-pack/plugin/identity-provider/build.gradle
+++ b/x-pack/plugin/identity-provider/build.gradle
@@ -45,7 +45,7 @@ dependencies {
   api "org.apache.httpcomponents:httpasyncclient:${versions.httpasyncclient}"
   api "org.apache.httpcomponents:httpcore-nio:${versions.httpcore}"
   api "org.apache.httpcomponents:httpclient-cache:${versions.httpclient}"
-  runtimeOnly 'com.google.guava:guava:19.0'
+  runtimeOnly 'com.google.guava:guava:32.1.1-jre'
 
   testImplementation "org.elasticsearch:mocksocket:${versions.mocksocket}"
   testImplementation(testArtifact(project(xpackModule('core'))))
