diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4f6f827..7b9fe43 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -122,6 +122,10 @@ if(${TRITON_ENABLE_GPU})
   enable_language(CUDA)
   include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/define.cuda_architectures.cmake)
   set_target_properties(kernel_library_new PROPERTIES LANGUAGE CUDA)
+  # Override CUDA_ARCHITECTURES if environment variable is set
+  if(DEFINED ENV{CUDA_ARCHITECTURES})
+    set(CUDA_ARCHITECTURES $ENV{CUDA_ARCHITECTURES})
+  endif()
   set_target_properties(kernel_library_new PROPERTIES CUDA_ARCHITECTURES "${CUDA_ARCHITECTURES}")
   set_target_properties(kernel_library_new PROPERTIES POSITION_INDEPENDENT_CODE ON)
   set_target_properties(kernel_library_new PROPERTIES LINKER_LANGUAGE CUDA)