diff --git a/CMakeLists.txt b/CMakeLists.txt
index f5c3356..742c421 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -121,7 +121,11 @@ if(${TRITON_ENABLE_GPU})
 
   enable_language(CUDA)
   set_target_properties(kernel_library_new PROPERTIES LANGUAGE CUDA)
-  set_target_properties(kernel_library_new PROPERTIES CUDA_ARCHITECTURES "75-real;80-real;86-real;87-real;89-real;90-real;100-real;101-real;120-real")
+  if(DEFINED ENV{CUDA_ARCHITECTURES})
+    set_target_properties(kernel_library_new PROPERTIES CUDA_ARCHITECTURES $ENV{CUDA_ARCHITECTURES})
+  else()
+    set_target_properties(kernel_library_new PROPERTIES CUDA_ARCHITECTURES "75-real;80-real;86-real;87-real;89-real;90-real;100-real;101-real;120-real")
+  endif()
   set_target_properties(kernel_library_new PROPERTIES POSITION_INDEPENDENT_CODE ON)
   set_target_properties(kernel_library_new PROPERTIES LINKER_LANGUAGE CUDA)
   target_compile_features(kernel_library_new PUBLIC cxx_std_${TRITON_MIN_CXX_STANDARD})