diff --git a/src/libtorch_utils.h b/src/libtorch_utils.h
index 6ec325b..0ac84bc 100644
--- a/src/libtorch_utils.h
+++ b/src/libtorch_utils.h
@@ -34,7 +34,6 @@
 #pragma GCC diagnostic ignored "-Wsign-compare"
 #pragma warning(push, 0)
 #include <torch/csrc/jit/codegen/cuda/interface.h>
-#include <torch/csrc/jit/codegen/fuser/interface.h>
 #include <torch/csrc/jit/passes/tensorexpr_fuser.h>
 #include <torch/csrc/jit/runtime/graph_executor.h>
 #include <torch/script.h>  // One-stop header for TorchScript
