From 3234a17cb82cd7f4aa4a78e022e9814215bacfb2 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Tue, 11 Mar 2025 11:32:12 +0000
Subject: [PATCH 2/2] Add support for apk installed db

---
 .../application/dataProvider_unix.go          | 43 +++++++++++++++++++
 1 file changed, 43 insertions(+)

diff --git a/agent/plugins/inventory/gatherers/application/dataProvider_unix.go b/agent/plugins/inventory/gatherers/application/dataProvider_unix.go
index d03ecc7e..d055d886 100644
--- a/agent/plugins/inventory/gatherers/application/dataProvider_unix.go
+++ b/agent/plugins/inventory/gatherers/application/dataProvider_unix.go
@@ -27,6 +27,7 @@ import (
 	"strings"
 	"time"
 
+	"chainguard.dev/apko/pkg/apk/apk"
 	"github.com/aws/amazon-ssm-agent/agent/context"
 	"github.com/aws/amazon-ssm-agent/agent/log"
 	"github.com/aws/amazon-ssm-agent/agent/platform"
@@ -66,6 +67,9 @@ var (
 
 	// platforms that can pass application inventory files, as the agent cannot gather the data from the local package manager
 	inventoryApplicationFileSupportedPlatforms = []string{"Bottlerocket"}
+
+	// platforms that have an apk-tools v2 installed db
+	apkToolsInstalledDB = "/lib/apk/db/installed"
 )
 
 func randomString(length int) string {
@@ -174,6 +178,19 @@ func collectPlatformDependentApplicationData(context context.T) (appData []model
 		}
 	}
 
+	if fileExists(apkToolsInstalledDB) {
+		noPackageManagerFound = false
+		var apkToolsAppData []model.ApplicationData
+		if apkToolsAppData, err = getApkToolsApplicationData(platformName); err != nil {
+			log.Errorf("Failed to gather inventory data for %v: %v", GathererName, err)
+		} else {
+			log.Infof("Appending application information found using ApkTools v2 Installed DB to application data.")
+			log.Infof("Found %v apk packages", len(apkToolsAppData))
+			appData = append(appData, apkToolsAppData...)
+
+		}
+	}
+
 	log.Infof("Found %v packages in total", len(appData))
 
 	if noPackageManagerFound {
@@ -237,6 +254,32 @@ func getInventoryApplicationFileData(inventoryApplicationFileBytes []byte) (data
 	return
 }
 
+// getApkToolsApplicationData creates ApplicationData from apk-tools
+// v2 installed DB
+func getApkToolsApplicationData(platformName string) (data []model.ApplicationData, err error) {
+	a, err := apk.New()
+	if err != nil {
+		return nil, err
+	}
+	pkgs, err := a.GetInstalled()
+	if err != nil {
+		return nil, err
+	}
+	for _, pkg := range pkgs {
+		var item model.ApplicationData
+		item.Name = pkg.Name
+		item.Version = pkg.Version
+		// Not quite right, but might be helpful
+		item.InstalledTime = pkg.BuildTime.Format(time.RFC3339)
+		item.Architecture = pkg.Arch
+		item.URL = pkg.URL
+		item.Summary = pkg.Description
+		item.PackageId = fmt.Sprintf("pkg:apk/%s/%s@%s?arch=%s&source=%s", platformName, pkg.Name, pkg.Version, pkg.Arch, pkg.Origin)
+		data = append(data, item)
+	}
+	return data, nil
+}
+
 // getApplicationData runs a shell command and gets information about all packages/applications
 func getApplicationData(context context.T, command string, args []string) (data []model.ApplicationData, err error) {
 
-- 
2.43.0

