diff --git a/unoconv b/unoconv
index bf67c9c..68f0fcf 100644
--- a/unoconv
+++ b/unoconv
@@ -1013,7 +1013,7 @@ class Convertor:
                 inputprops += UnoProps(InputStream=inputStream)
                 inputurl = 'private:stream'
             elif os.path.exists(inputfn):
-                inputurl = unohelper.absolutize(self.cwd, unohelper.systemPathToFileUrl(inputfn))
+                inputurl = uno.absolutize(self.cwd, unohelper.systemPathToFileUrl(inputfn))
             else:
                 inputurl = inputfn
             document = self.desktop.loadComponentFromURL(inputurl, "_blank", 0, inputprops)
@@ -1027,7 +1027,7 @@ class Convertor:
                 if os.path.exists(op.template):
                     info(1, "Template file: %s" % op.template)
                     templateprops = UnoProps(OverwriteStyles=True)
-                    templateurl = unohelper.absolutize(self.cwd, unohelper.systemPathToFileUrl(op.template))
+                    templateurl = uno.absolutize(self.cwd, unohelper.systemPathToFileUrl(op.template))
                     document.StyleFamilies.loadStylesFromURL(templateurl, templateprops)
                 else:
                     print('unoconv: template file `%s\' does not exist.' % op.template, file=sys.stderr)
@@ -1168,7 +1168,7 @@ class Convertor:
                 else:
                     outputfn = realpath(inbase + os.extsep + outputfmt.extension)
 
-                outputurl = unohelper.absolutize(self.cwd, unohelper.systemPathToFileUrl(outputfn))
+                outputurl = uno.absolutize(self.cwd, unohelper.systemPathToFileUrl(outputfn))
 
             info(1, "Output file: %s" % outputurl)
 
