diff --git a/Gemfile b/Gemfile
index 9dc6f14926d..583d8cdd1a6 100644
--- a/Gemfile
+++ b/Gemfile
@@ -147,7 +147,7 @@ gem 'browser', '~> 5.3.1', feature_category: :shared
 gem 'ohai', '~> 18.1', feature_category: :product_analytics
 
 # GPG
-gem 'gpgme', '~> 2.0.24', feature_category: :source_code_management
+gem 'gpgme', '~> 2.0.25', feature_category: :source_code_management
 
 # LDAP Auth
 # GitLab fork with several improvements to original library. For full list of changes
diff --git a/Gemfile.checksum b/Gemfile.checksum
index 3f9851ac831..55c16fbe02f 100644
--- a/Gemfile.checksum
+++ b/Gemfile.checksum
@@ -283,7 +283,6 @@
 {"name":"googleapis-common-protos","version":"1.4.0","platform":"ruby","checksum":"da2380fb5ab1563580816c74e8d684ac17512c3654c829a3ee84f6d6139de382"},
 {"name":"googleapis-common-protos-types","version":"1.20.0","platform":"ruby","checksum":"5e374b06bcfc7e13556e7c0d87b99f1fa3d42de6396a1de3d8fc13aefb4dd07f"},
 {"name":"googleauth","version":"1.8.1","platform":"ruby","checksum":"814adadaaa1221dce72a67131e3ecbd6d23491a161ec84fb15fd353b87d8c9e7"},
-{"name":"gpgme","version":"2.0.24","platform":"ruby","checksum":"53eccd7042abb4fd5c78f30bc9ed075b1325e6450eab207f2f6a1e7e28ae3b64"},
 {"name":"grape","version":"2.0.0","platform":"ruby","checksum":"3aeff94c17e84ccead4ff98833df691e7da0c108878cc128ca31f80c1047494a"},
 {"name":"grape-entity","version":"1.0.1","platform":"ruby","checksum":"e00f9e94e407aff77aa2945d741f544d07e48501927942988799913151d02634"},
 {"name":"grape-path-helpers","version":"2.0.1","platform":"ruby","checksum":"ad5216e52c6e796738a9118087352ab4c962900dbad1d8f8c0f96e093c6702d7"},
diff --git a/Gemfile.lock b/Gemfile.lock
index fa3c6928178..6042f70852c 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -918,7 +918,7 @@ GEM
       multi_json (~> 1.11)
       os (>= 0.9, < 2.0)
       signet (>= 0.16, < 2.a)
-    gpgme (2.0.24)
+    gpgme (2.0.25)
       mini_portile2 (~> 2.7)
     grape (2.0.0)
       activesupport (>= 5)
@@ -2194,7 +2194,7 @@ DEPENDENCIES
   google-cloud-storage (~> 1.45.0)
   google-protobuf (~> 3.25, >= 3.25.3)
   googleauth (~> 1.8.1)
-  gpgme (~> 2.0.24)
+  gpgme (~> 2.0.25)
   grape (~> 2.0.0)
   grape-entity (~> 1.0.1)
   grape-path-helpers (~> 2.0.1)
diff --git a/Gemfile.next.checksum b/Gemfile.next.checksum
index 737e7c4c652..2a9b8b4db32 100644
--- a/Gemfile.next.checksum
+++ b/Gemfile.next.checksum
@@ -283,7 +283,6 @@
 {"name":"googleapis-common-protos","version":"1.4.0","platform":"ruby","checksum":"da2380fb5ab1563580816c74e8d684ac17512c3654c829a3ee84f6d6139de382"},
 {"name":"googleapis-common-protos-types","version":"1.20.0","platform":"ruby","checksum":"5e374b06bcfc7e13556e7c0d87b99f1fa3d42de6396a1de3d8fc13aefb4dd07f"},
 {"name":"googleauth","version":"1.8.1","platform":"ruby","checksum":"814adadaaa1221dce72a67131e3ecbd6d23491a161ec84fb15fd353b87d8c9e7"},
-{"name":"gpgme","version":"2.0.24","platform":"ruby","checksum":"53eccd7042abb4fd5c78f30bc9ed075b1325e6450eab207f2f6a1e7e28ae3b64"},
 {"name":"grape","version":"2.0.0","platform":"ruby","checksum":"3aeff94c17e84ccead4ff98833df691e7da0c108878cc128ca31f80c1047494a"},
 {"name":"grape-entity","version":"1.0.1","platform":"ruby","checksum":"e00f9e94e407aff77aa2945d741f544d07e48501927942988799913151d02634"},
 {"name":"grape-path-helpers","version":"2.0.1","platform":"ruby","checksum":"ad5216e52c6e796738a9118087352ab4c962900dbad1d8f8c0f96e093c6702d7"},
diff --git a/Gemfile.next.lock b/Gemfile.next.lock
index 553c2b2aada..902b71b9906 100644
--- a/Gemfile.next.lock
+++ b/Gemfile.next.lock
@@ -912,7 +912,7 @@ GEM
       multi_json (~> 1.11)
       os (>= 0.9, < 2.0)
       signet (>= 0.16, < 2.a)
-    gpgme (2.0.24)
+    gpgme (2.0.25)
       mini_portile2 (~> 2.7)
     grape (2.0.0)
       activesupport (>= 5)
@@ -2189,7 +2189,7 @@ DEPENDENCIES
   google-cloud-storage (~> 1.45.0)
   google-protobuf (~> 3.25, >= 3.25.3)
   googleauth (~> 1.8.1)
-  gpgme (~> 2.0.24)
+  gpgme (~> 2.0.25)
   grape (~> 2.0.0)
   grape-entity (~> 1.0.1)
   grape-path-helpers (~> 2.0.1)
