From 481f652907d0e0ca3eb422b37cb36e4d22421fef Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Mon, 9 Jun 2025 16:30:26 +0100
Subject: [PATCH 1/2] version: Print FIPS module version

---
 source/common/version/version.cc | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/source/common/version/version.cc b/source/common/version/version.cc
index cf671be189..add19fd36a 100644
--- a/source/common/version/version.cc
+++ b/source/common/version/version.cc
@@ -50,7 +50,12 @@ const std::string& VersionInfo::buildType() {
 
 const std::string& VersionInfo::sslVersion() {
 #ifdef ENVOY_SSL_VERSION
-  static const std::string ssl_version = ENVOY_SSL_VERSION;
+  #if defined OPENSSL_IS_BORINGSSL || defined OPENSSL_IS_AWSLC
+    static const std::string ssl_version = fmt::format("{}-{}", ENVOY_SSL_VERSION,
+                                                       FIPS_version());
+  #else
+    static const std::string ssl_version = ENVOY_SSL_VERSION;
+  #endif
 #else
   static const std::string ssl_version = "no-ssl";
 #endif
-- 
2.48.1

