From 08ec22097ec1c8642d43122a2e25eb28d543939f Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Fri, 23 Jan 2026 22:45:34 +0000
Subject: [PATCH] fips: always set ssl_compliance_policy_fips_202205

---
 source/common/tls/context_impl.cc | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/source/common/tls/context_impl.cc b/source/common/tls/context_impl.cc
index 283a5c0232..4f6bef530c 100644
--- a/source/common/tls/context_impl.cc
+++ b/source/common/tls/context_impl.cc
@@ -362,14 +362,6 @@ ContextImpl::ContextImpl(
     }
   }
 
-  // Compliance policy must be applied last to have a defined behavior.
-  if (const auto policy = config.compliancePolicy(); policy.has_value()) {
-    switch (policy.value()) {
-      using ProtoPolicy = envoy::extensions::transport_sockets::tls::v3::TlsParameters;
-    case ProtoPolicy::FIPS_202205:
-      if (!fips_mode) {
-        ENVOY_LOG(warn, "FIPS conformance policy applied on a non-FIPS build");
-      }
       for (auto& tls_context : tls_contexts_) {
         int rc = SSL_CTX_set_compliance_policy(tls_context.ssl_ctx_.get(),
                                                ssl_compliance_policy_fips_202205);
@@ -380,12 +372,6 @@ ContextImpl::ContextImpl(
           return;
         }
       }
-      break;
-    default:
-      creation_status = absl::InvalidArgumentError("Unknown compliance policy");
-      return;
-    }
-  }
 }
 
 void ContextImpl::keylogCallback(const SSL* ssl, const char* line) {
-- 
2.51.0

