diff --git a/proxy.js b/proxy.js
index b78c94a..12e1da5 100644
--- a/proxy.js
+++ b/proxy.js
@@ -92,7 +92,7 @@ configlib.configFile(process.argv[2], function (conf, oldConfig) {
   });
 
   var ring = new hashring(
-    node_ring, 'md5', {
+    node_ring, 'sha2-256', {
       'max cache size': config.cacheSize || 10000,
       //We don't want duplicate keys sent so replicas set to 0
       'replicas': 0
