diff --git a/pkg/bootstrap/bootstrap.go b/pkg/bootstrap/bootstrap.go
index 75263176..b6c9583f 100644
--- a/pkg/bootstrap/bootstrap.go
+++ b/pkg/bootstrap/bootstrap.go
@@ -44,7 +44,7 @@ var (
 
 // binDirForDigest returns the path to dataDir/data/refDigest/bin.
 func binDirForDigest(dataDir string, refDigest string) string {
-	return filepath.Join(dataDir, "data", refDigest, "bin")
+	return filepath.Join(dataDir, "data", refDigest, "usr", "bin")
 }
 
 // chartsDirForDigest returns the path to dataDir/data/refDigest/charts.
@@ -65,7 +65,7 @@ func imagesDir(dataDir string) string {
 // symlinkBinDir returns the path to dataDir/bin.
 // This will be symlinked to the current runtime bin dir.
 func symlinkBinDir(dataDir string) string {
-	return filepath.Join(dataDir, "bin")
+	return filepath.Join(dataDir,  "usr", "bin")
 }
 
 // dirExists returns true if a directory exists at the given path.
@@ -141,7 +141,7 @@ func Stage(resolver *images.Resolver, nodeConfig *daemonconfig.Node, cfg cmds.Ag
 
 		// Extract binaries and charts
 		extractPaths := map[string]string{
-			"/bin":    refBinDir,
+			"/usr/bin":    refBinDir,
 			"/charts": refChartsDir,
 		}
 		if err := extract.ExtractDirs(img, extractPaths); err != nil {
