diff --git a/src/etc/systemd/resolved.conf.d/gce-resolved.conf b/src/etc/systemd/resolved.conf.d/gce-resolved.conf
index b2d4120..fd408e2 100644
--- a/src/etc/systemd/resolved.conf.d/gce-resolved.conf
+++ b/src/etc/systemd/resolved.conf.d/gce-resolved.conf
@@ -20,4 +20,6 @@ MulticastDNS=false
 # Systemd-resolved will still pick the right server from DHCP, but if we don't
 # explicilty configure it here we can't use it for .local domains.
 DNS=169.254.169.254
-
+# Disable DNSSEC which, if enabled, will fail to resolve DNS on gvnic machine
+# types on any distribution.
+DNSSEC=no
-- 
2.48.1

