diff --git a/x-pack/plugin/security/build.gradle b/x-pack/plugin/security/build.gradle
index 08496060f43..e5ee3312f4b 100644
--- a/x-pack/plugin/security/build.gradle
+++ b/x-pack/plugin/security/build.gradle
@@ -80,7 +80,7 @@ dependencies {

   // Dependencies for oidc
   api "com.nimbusds:oauth2-oidc-sdk:9.37"
-  api "com.nimbusds:nimbus-jose-jwt:9.23"
+  api "com.nimbusds:nimbus-jose-jwt:9.37.2"
   api "com.nimbusds:lang-tag:1.4.4"
   api "com.sun.mail:jakarta.mail:1.6.3"
   api "net.jcip:jcip-annotations:1.0"
