diff --git a/x-pack/plugin/core/build.gradle b/x-pack/plugin/core/build.gradle
index 71dd5bed6cf..aa234181739 100644
--- a/x-pack/plugin/core/build.gradle
+++ b/x-pack/plugin/core/build.gradle
@@ -49,7 +49,7 @@ dependencies {
 
   // security deps
   api 'com.unboundid:unboundid-ldapsdk:6.0.3'
-  api "com.nimbusds:nimbus-jose-jwt:9.23"
+  api "com.nimbusds:nimbus-jose-jwt:9.37.2"
 
   implementation project(":x-pack:plugin:core:template-resources")