diff --git a/gradle/build.versions.toml b/gradle/build.versions.toml
index bbcad622cf5..e965780242b 100644
--- a/gradle/build.versions.toml
+++ b/gradle/build.versions.toml
@@ -7,7 +7,7 @@ spock = "2.1-groovy-3.0"
 [libraries]
 ant = "org.apache.ant:ant:1.10.12"
 antlrst4 = "org.antlr:ST4:4.3.4"
-apache-compress = "org.apache.commons:commons-compress:1.24.0"
+apache-compress = "org.apache.commons:commons-compress:1.26.0"
 apache-rat = "org.apache.rat:apache-rat:0.11"
 asm = { group = "org.ow2.asm", name="asm", version.ref="asm" }
 asm-tree = { group = "org.ow2.asm", name="asm-tree", version.ref="asm" }
