diff --git a/modules/ingest-attachment/build.gradle b/modules/ingest-attachment/build.gradle
index 2c15ea076e11..323203170f77 100644
--- a/modules/ingest-attachment/build.gradle
+++ b/modules/ingest-attachment/build.gradle
@@ -86,7 +86,7 @@ dependencies {
   // MS Office
   api "org.apache.poi:poi-scratchpad:${versions.poi}"
   // Apple iWork
-  api 'org.apache.commons:commons-compress:1.24.0'
+  api 'org.apache.commons:commons-compress:1.26.1'
   // Outlook documents
   api "org.apache.james:apache-mime4j-core:${versions.mime4j}"
   api "org.apache.james:apache-mime4j-dom:${versions.mime4j}"
