diff --git a/modules/ingest-attachment/build.gradle b/modules/ingest-attachment/build.gradle
index b10f3ac3195..07de0fa4301 100644
--- a/modules/ingest-attachment/build.gradle
+++ b/modules/ingest-attachment/build.gradle
@@ -21,7 +21,7 @@ def versions = [
   'tika'  : '2.7.0',
   'pdfbox': '2.0.27',
   'poi'   : '5.2.3',
-  'mime4j': '0.8.9'
+  'mime4j': '0.8.10'
 ]
 
 // exclude commons-logging from test dependencies to avoid jar-hell, we use jcl-over-slf4j here
@@ -76,7 +76,7 @@ dependencies {
   // MS Office
   api "org.apache.poi:poi-scratchpad:${versions.poi}"
   // Apple iWork
-  api 'org.apache.commons:commons-compress:1.24.0'
+  api 'org.apache.commons:commons-compress:1.26.0'
   // Outlook documents
   api "org.apache.james:apache-mime4j-core:${versions.mime4j}"
   api "org.apache.james:apache-mime4j-dom:${versions.mime4j}"
