diff --git a/test/x-content/build.gradle b/test/x-content/build.gradle
index 9c00e32b413..4ab9718717c 100644
--- a/test/x-content/build.gradle
+++ b/test/x-content/build.gradle
@@ -18,7 +18,7 @@ dependencies {
   implementation "com.networknt:json-schema-validator:${versions.networknt_json_schema_validator}"
   implementation "com.fasterxml.jackson.core:jackson-databind:${versions.jackson}"
   implementation "com.fasterxml.jackson.core:jackson-annotations:${versions.jackson}"
-  implementation "org.apache.commons:commons-compress:1.24.0"
+  implementation "org.apache.commons:commons-compress:1.26.0"
   implementation "org.apache.commons:commons-lang3:${versions.commons_lang3}"
 }
 
