From 9090b67dfedf46c3e25d228339f81d67c6b020ef Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Wed, 31 Dec 2025 13:08:06 +0000
Subject: [PATCH] util: underlinked dns_query

---
 src/mongo/util/BUILD.bazel | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/mongo/util/BUILD.bazel b/src/mongo/util/BUILD.bazel
index 918735f5a01..8865892fee2 100644
--- a/src/mongo/util/BUILD.bazel
+++ b/src/mongo/util/BUILD.bazel
@@ -419,6 +419,9 @@ mongo_cc_library(
         "//bazel/config:posix": ["dns_query_posix-impl.h"],
         "//conditions:default": [],
     }),
+    linkopts = [
+        "-lresolv",
+    ],
     deps = [
         "//src/mongo:base",
     ],
-- 
2.51.0

