Origin: https://salsa.debian.org/multimedia-team/libmad/-/blob/debian/0.15.1b-13/debian/patches/64bit-targets.diff?ref_type=tags

Index: libmad-0.15.1b/configure.ac
===================================================================
--- libmad-0.15.1b.orig/configure.ac	2008-03-07 20:33:05.000000000 +0000
+++ libmad-0.15.1b/configure.ac	2024-06-19 08:32:03.000000000 +0000
@@ -233,6 +233,8 @@
 then
     case "$host" in
 	i?86-*)     FPM="INTEL"  ;;
+	x86_64*)    FPM="64BIT"  ;;
+	aarch64*)   FPM="64BIT"  ;;
 	arm*-*)     FPM="ARM"    ;;
 	mips*-*)    FPM="MIPS"   ;;
 	sparc*-*)   FPM="SPARC"  ;;
