From 5939e30af4e2268340620a8833d8d2285eda485a Mon Sep 17 00:00:00 2001
From: Arturo Borrero Gonzalez <arturo.bg@arturo.bg>
Date: Tue, 28 Oct 2025 16:58:08 +0100
Subject: [PATCH] link ordering

Signed-off-by: Arturo Borrero Gonzalez <arturo.bg@arturo.bg>
---
 dcgmi/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dcgmi/CMakeLists.txt b/dcgmi/CMakeLists.txt
index 3697670..0ad41c7 100755
--- a/dcgmi/CMakeLists.txt
+++ b/dcgmi/CMakeLists.txt
@@ -61,11 +61,11 @@ target_link_libraries(dcgmi
     PRIVATE
         buildinfo_objects
         common_interface
-        dcgm
         dcgm_interface
         dcgm_mutex
         dcgm_thread
         dcgmi_objects
+        dcgm
         fmt::fmt
         nvvs_interface)
 
-- 
2.47.3

