diff --git a/Sources/Build/BuildDescription/ClangModuleBuildDescription.swift b/Sources/Build/BuildDescription/ClangModuleBuildDescription.swift
index 15f5290f8..2388038a3 100644
--- a/Sources/Build/BuildDescription/ClangModuleBuildDescription.swift
+++ b/Sources/Build/BuildDescription/ClangModuleBuildDescription.swift
@@ -406,7 +406,7 @@ public final class ClangModuleBuildDescription {
         case .debug:
             return ["-O0"]
         case .release:
-            return ["-O2"]
+            return ["-O1"]
         }
     }
 
