--- swift-project.orig/llvm-project/llvm/include/llvm/ADT/SmallVector.h
+++ swift-project/llvm-project/llvm/include/llvm/ADT/SmallVector.h
@@ -19,6 +19,7 @@
 #include <algorithm>
 #include <cassert>
 #include <cstddef>
+#include <cstdint>
 #include <cstdlib>
 #include <cstring>
 #include <functional>
--- swift-project.orig/llvm-project/llvm/lib/Target/X86/MCTargetDesc/X86MCTargetDesc.h
+++ swift-project/llvm-project/llvm/lib/Target/X86/MCTargetDesc/X86MCTargetDesc.h
@@ -13,6 +13,7 @@
 #ifndef LLVM_LIB_TARGET_X86_MCTARGETDESC_X86MCTARGETDESC_H
 #define LLVM_LIB_TARGET_X86_MCTARGETDESC_X86MCTARGETDESC_H

+#include <cstdint>
 #include <memory>
 #include <string>

--- swift-project.orig/llvm-project/third-party/unittest/googletest/src/gtest-death-test.cc
+++ swift-project/llvm-project/third-party/unittest/googletest/src/gtest-death-test.cc
@@ -32,6 +32,7 @@

 #include "gtest/gtest-death-test.h"

+#include <cstdint>
 #include <utility>

 #include "gtest/internal/gtest-port.h"
--- swift-project.orig/swift/stdlib/include/llvm/ADT/SmallVector.h
+++ swift-project/swift/stdlib/include/llvm/ADT/SmallVector.h
@@ -21,6 +21,7 @@
 #include <algorithm>
 #include <cassert>
 #include <cstddef>
+#include <cstdint>
 #include <cstdlib>
 #include <cstring>
 #include <functional>
