From bcd1fb62ec6182362d136bce5db0f25d773ff54a Mon Sep 17 00:00:00 2001
From: Ajay K <ajaykemparaj@gmail.com>
Date: Wed, 5 Mar 2025 23:35:57 +0000
Subject: [PATCH 04/11] default go fips enabled

---
 src/internal/buildcfg/cfg.go | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/src/internal/buildcfg/cfg.go b/src/internal/buildcfg/cfg.go
index f6d6a78531..69d9941f1c 100644
--- a/src/internal/buildcfg/cfg.go
+++ b/src/internal/buildcfg/cfg.go
@@ -106,16 +106,7 @@ func goamd64() int {
 }
 
 func gofips140() string {
-	v := envOr("GOFIPS140", DefaultGOFIPS140)
-	switch v {
-	case "off", "latest", "inprocess", "certified":
-		return v
-	}
-	if isFIPSVersion(v) {
-		return v
-	}
-	Error = fmt.Errorf("invalid GOFIPS140: must be off, latest, inprocess, certified, or v1.Y.Z")
-	return DefaultGOFIPS140
+	return "latest"
 }
 
 // isFIPSVersion reports whether v is a valid FIPS version,
-- 
2.51.0

