From 5ae0e279a86cee550c0098db00a4f98795f3bcd4 Mon Sep 17 00:00:00 2001
From: Ajay K <ajaykemparaj@gmail.com>
Date: Thu, 6 Mar 2025 04:31:06 +0000
Subject: [PATCH 05/11] set gofips as latest always

---
 src/cmd/dist/build.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cmd/dist/build.go b/src/cmd/dist/build.go
index 5dedec8802..a3d6739a13 100644
--- a/src/cmd/dist/build.go
+++ b/src/cmd/dist/build.go
@@ -179,7 +179,7 @@ func xinit() {
 
 	b = os.Getenv("GOFIPS140")
 	if b == "" {
-		b = "off"
+		b = "latest"
 	}
 	gofips140 = b
 
-- 
2.51.0

