commit 87ca5837ac04919071e227dbdb07fb08999fb54d (HEAD)
Author: Melange Build <melange-build@cgr.dev>
Date:   Tue Aug 19 14:29:35 2025 +0000

    Bump version to remediate GHSA-4cx2-fc23-5wg6

diff --git a/gradle/libs.versions.toml b/gradle/libs.versions.toml
index b699d475..338b5f39 100644
--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -56,7 +56,7 @@ reactivestreams   = "1.0.4"
 # when updating this version, you need to ensure compatibility with:
 #  - plugins/ingest-attachment (transitive dependency, check the upstream POM)
 #  - distribution/tools/plugin-cli
-bouncycastle="1.78"
+bouncycastle="1.79"
 # test dependencies
 randomizedrunner  = "2.7.1"
 junit             = "4.13.2"
