diff --git a/airflow/lineage/hook.py b/airflow/lineage/hook.py
index 9aeb65c..ac86e76 100644
--- a/airflow/lineage/hook.py
+++ b/airflow/lineage/hook.py
@@ -97,7 +97,7 @@ class HookLineageCollector(LoggingMixin):
         unique for each combination of asset and context.
         """
         extra_str = json.dumps(asset.extra, sort_keys=True)
-        extra_hash = hashlib.md5(extra_str.encode()).hexdigest()
+        extra_hash = hashlib.md5(extra_str.encode(), usedforsecurity=False).hexdigest()
         return f"{asset.uri}_{extra_hash}_{id(context)}"
 
     def create_asset(
diff --git a/airflow/models/dagbag.py b/airflow/models/dagbag.py
index 393d01c..3c14d4c 100644
--- a/airflow/models/dagbag.py
+++ b/airflow/models/dagbag.py
@@ -669,7 +669,7 @@ class DagBag(LoggingMixin):
 def generate_md5_hash(context):
     bundle_name = context.get_current_parameters()["bundle_name"]
     relative_fileloc = context.get_current_parameters()["relative_fileloc"]
-    return hashlib.md5(f"{bundle_name}:{relative_fileloc}".encode()).hexdigest()
+    return hashlib.md5(f"{bundle_name}:{relative_fileloc}".encode(), usedforsecurity=False).hexdigest()
 
 
 class DagPriorityParsingRequest(Base):
